

Cette méthode suppose qu'il existe une structure résiduelle à l'état déplié, avec certains
Les positions sont plus enterrées que d'autres. En outre, il devrait faire le paramétrage
Plus robuste et moins sensible à la taille et à la structure des homologues naturels utilisés pour
Définissent les compositions d'acides aminés cibles, car les fréquences d'acides aminés des exposés
Et les régions enterrées sont calculées en moyenne séparément. En principe, cela double le nombre de
Énergies de référence réglables. Cependant, nous avons réduit ce nombre en introduisant des amino
Classes de similarité acide, avec une énergie de référence réglable par classe. Pour optimiser
Énergie de référence, nous avons effectué des calculs de conception pour chaque protéine d'essai (état de l'apo)
Où la moitié des positions amino pourraient muter à la fois (à l'exclusion de Gly et Pro), avec
Des simulations distinctes pour chaque moitié. De cette façon, lors de l'optimisation des paramètres, une
La position était toujours entourée d'un environnement au moins 50% identique au type sauvage
séquence. Les calculs de conception s'appuyaient sur une puissante et efficace réplique d'échange
Méthode d'exploration de Monte Carlo qui a utilisé plus d'un demi-milliard de pas par simulation (par
Réplique), et a produit des milliers de séquences conçues dans une seule simulation. Référence
Les valeurs d'énergie ont été optimisées avec deux choix différents pour la constante diélectrique des protéines
P. Les niveaux de performance étaient similaires aux deux valeurs.
Le modèle présente plusieurs limitations, dont la plupart sont répandues dans les implémentations et les applications de CPD. Le premier est l'utilisation de la stabilité des protéines comme seul critère de conception,
Sans expliquer explicitement la spécificité du pli 77, la protection contre l'agrégation, ou des considérations fonctionnelles comme la liaison des ligands. Nous notons toutefois que les tests Superfamily
N'a pas entraîné de mauvaises affections (séquences qui préfèrent un autre pli SCOP), donc
Qu'en pratique, la spécificité du pli a été réalisée. Des critères fonctionnels peuvent également être introduits
D'une manière ad hoc; Par exemple, les séquences testées par MD ont été conçues avec 11 peptides
Résidus contraignants fixés, pour faciliter les futures études expérimentales.
Une deuxième limitation du modèle est l'utilisation d'un squelette protéique fixe. En fait, l'épine dorsale
N'est pas vraiment réparé: plutôt, certains mouvements sont autorisés, mais modélisés implicitement, à travers les
Utilisation d'une constante diélectrique protéique supérieure à une (P = 4 ou 8) 78. Cette valeur diélectrique
Signifie que la structure protéique (y compris son squelette) est autorisée à se détendre ou à se réorganiser en réponse à une redistribution de charge associée à des mutations ou à un rotarymide latéral
changements. Cependant, la réorganisation est modélisée implicitement, pas explicitement 78, et elle ne l'est pas
Ne pas impliquer le mouvement des centres atomiques ou de leurs sphères associées van der Waals. Ainsi,
L'épine dorsale ne peut pas se réorganiser en réponse à une répulsion stérique produite par des mutations ou
26

Les changements de rotérares, et ils ne peuvent pas non plus se déplacer pour remplir l'espace laissé vide par une mutation. L'effet de cette
L'approximation était apparente dans la conception des quatre positions de spécificité Tiam1, où
Les séquences conçues étaient sensibles à la conformation du squelette particulier de la protéine et du peptide. Plus précisément, avec la structure du squelette de type sauvage, il n'y avait pas de place
Pour insérer une chaîne latérale Phe à la position 915, même si Phe915 est présent dans le mutant quadruple expérimental (qui a une structure de squelette légèrement différente). Donc,
Le choix du modèle structurel de rayons X initial est important et plusieurs stratégies sont
possible. Ici, pour paramétrer le modèle CPD, nous avons utilisé des structures de rayons X résolues avec
Un ligand peptidique, même si les simulations de paramétrage et la plupart des tests
Ont été effectués pour les protéines apo. Ce choix a été fait en partie parce que l'apo / holo PDZ
Les structures sont assez similaires et en partie pour rendre le modèle plus transférable et faciliter
Applications à la liaison peptidique. Une autre stratégie aurait pu être de paramétrer les
Modèle en utilisant toutes les structures apo, puis passer aux structures holo pour l'application Tiam1.
Pour la conception complète des protéines (comme la demande de titrage hydrophobe), l'utilisation d'un
Le squelette fixe peut être en partie contrebalancé en concevant deux ou plusieurs structures PDZ.
Par exemple, la mise en commun des séquences Tiam1 et Cask conçues a donné une séquence moyenne
Entropie comparable à l'ensemble expérimental Pfam, et nous a permis de récapituler plus
Séquences que de conception avec un seul squelette. Dans l'application à Tiam1 4-position
Conception, le squelette fixe a également été contrebalancé en effectuant des calculs séparément avec
Deux structures de squelette différentes, un holo wildtype et une structure de holo mutant. Les simulations avec le squelette mutant nous ont permis d'obtenir des mutants ayant Phe à la position 915.
Notez qu'une nouvelle méthode pour la conception multibackbone a récemment été développée dans Proteus,
Sur la base d'une méthode hybride hybride non-heuristique novatrice qui préserve Boltzmann
Échantillonnage 35. Cette méthode pourrait être appliquée à l'avenir.
Une troisième limitation de notre modèle est la nécessité, pour des résultats optimaux, de paramétrer les
Énergies de référence spécifiquement pour un ensemble donné de protéines. Cette étape est bien automatisée
Et très parallèle. Cependant, cela implique plusieurs choix qui sont partiellement arbitraires. Celles-ci
Comprennent le choix d'un ensemble de domaines protéiques pour représenter la protéine ou la famille d'intérêt.
Nous devons également choisir un seuil de similarité pour définir les homologues cibles à partir desquels
On calcule les compositions expérimentales d'acides aminés. Ici, nous avons choisi d'utiliser
Les homologues de chaque membre de la famille, calculent leurs compositions, puis la moyenne sur les deux
Représentants familiaux. Cette méthode a fonctionné bien, mais d'autres choix sont possibles, et plus encore
27

Il faut travailler pour tirer des conclusions définitives. Also, the mono-position design of Tiam1
showed evidence of some systematic error (Fig. 4), with a large fraction of Arg, Lys, and
Gln residues types on the protein surface, despite the optimized reference energies. Dans
the future, it may be necessary to relax the intra-group constraints towards the end of
the reference energy optimization and/or target smaller numbers of mutating positions,
instead of one half of the protein at a time.
A fourth limitation of our model is the discrete rotamer approximation, which requires
some adaptation of the energy function to avoid exaggerated steric clashes; la méthode
used here is the residue-pair minimization method described earlier 26,76 . A fifth limitation
is the use of a pairwise additive solvation model (as in most CPD models). Specifically,
the dielectric environment of each residue pair is assumed here to be native-like (so-called
“Native Environment approximation” or NEA 74,76 ). This leads to an energy function that
has the form of a sum over pairs of residues and that can be pre-calculated and stored
in an energy matrix, which then serves as a lookup table during the subsequent Monte
Carlo simulations. Despite this approximation, the model gave good results for a large
acid/base benchmark, a problem that is very sensitive to the electrostatic treatment 74 .
Some of these limitations could be removed in future work. In particular, since the
energy function is mostly physics-based, it can benefit rapidly from ongoing improvements in protein force fields and solvation models. Thus, the NEA approximation could
be removed in the future due to the recent implementation (manuscript in preparation)
of a more exact Generalized Born calculation, whose efficiency is comparable to the pairwise approximation 79 . We have also implemented an improved model for hydrophobic
solvation 80 , which is faster and more accurate than our current surface area energy term
(manuscript in preparation).

5.2

Model testing and applications

Designed sequences were extensively compared to natural sequences, through fold recognition tests, similarity calculations, and entropy calculations. In the test simulations, we
designed the entire protein sequence, so that all positions (except Gly and Pro) could
mutate freely, subject only to an overall bias towards the mean, experimental amino acid
composition (through the reference energies). Despite the lack of experimental bias or
28

constraints, the resulting sequences had a high overall similarity to the natural, Pfam sequences, as measured by the Blosum40 similarity scores. The scores obtained were mostly
comparable to the similarity scores between pairs of Pfam sequences themselves. Thus,
the sequences designed with Proteus resemble moderately-distant natural homologs. le
similarity was very strong for residues in the core of the protein, as observed in previous
CPD studies 30,72 . In contrast, for residues at the protein surface, similarity scores were
close to zero, the score one would obtain if one picked amino acid types randomly. Notice that many surface residues are involved in functional interactions, such as the eleven
peptide-binding residues in PDZ domains. Surface residues are also selected by evolution
to avoid aggregation or unwanted adhesion. Most of these functional constraints are not
explicitly accounted for in our design protocol (although the energy function indirectly
favors protein solubility). Despite the limited similarity scores for surface regions, fold
recognition with the Superfamily tool and the best design models was almost perfect.
Earlier fold recognition tests that used a simpler energy function gave a lower fold recognition rate of about 85% (for a larger and more diverse test set) and lower similarities 15,50 .
Evidently, the combined use of an improved protein force field, Generalized Born solvent,
and family-specific reference energies leads to designed sequences that are more native-like
and presumably better.
The Proteus sequences were also compared to sequences designed with the Rosetta
software (using default parameters), which has itself been extensively tested. Based on the
Blosum similarity scores (vs. natural sequences in Pfam) and the fold recognition tests,
the Proteus and Rosetta sequences appear to be of about the same quality. cependant,
Rosetta makes fewer mutations than Proteus, so that the identity scores, compared to
the corresponding wildtype protein, are about 6 percent points higher. This means that
Proteus mutates about five more positions, on average, per PDZ domain. This number
could easily be reduced, by adding to the Proteus energy function an explicit bias energy
term that increases with the number of mutations (away from the wildtype sequence). Un
equivalent bias energy was used above for just two simulations of Tiam1 and Cask (see
the “biased Proteus” results in the two upper panels of Fig. 5), to illustrate the possibility
of using experimentally-restrained sampling. It remains to be seen whether a restraint
based on the identity score would lead to more stable and realistic designed sequences.
Another attractive route for testing designed sequences is through high-level MD sim29

mulations. Here, ten designed Tiam1 sequences were tested in rather long MD simulations,
in the apo form, using the same protein force field as in the CPD model (Amber force
field) and an established explicit water model. These sequences were designed using Proteus, with Gly, Pro, and eleven peptide-binding positions held fixed but all others free to
mutate. Seven of the sequences remained stable over 200 ns simulation lengths and two
were extended up to a microsecond, which represents a very stringent test of the designs.
Sequence 6 was stable over the whole microsecond. Its mean deviation from the starting,
experimental wildtype structure was 1 Å, which is the same as the mean deviation in the
MD simulation of the wildtype sequence itself. The deviation between the mean sequence
6 MD structure and the mean wildtype MD structure was also small, just 1.2 Å. Sequence
2 was also simulated and remained stable until just before the end of the microsecond
simulation, at which point it underwent a larger fluctuation. The fluctuation regressed
100 ns later. An even longer simulation would be needed to determine if this fluctuation
is harmless or signals the beginning of domain unfolding. Note that in the presence of a
peptide ligand, we expect the structural stability of the designed domains would increase
further. MD simulations of additional designed sequences would also be of interest. Direct
experimental testing of the designed proteins remains to be done.
The CPD model was used for two applications. “Hydrophobic titration” of two
PDZ domains illustrated a novel way to characterize protein designability. The cost or
availability of hydrophobic sidechain types was controlled by a bias energy term that
was gradually varied. One result was that the mean overall hydrophobic “susceptibility”,
the number of type changes per kcal/mol and per position, differed by a factor of two
between Tiam1 and Cask. In Tiam1, 12 of the core positions remained hydrophobic
even with the largest bias value favoring polar types, while 17 other positions changed to
polar (respectively, nonpolar) types at the largest polar (nonpolar) bias energy values. UNE
comparison to other domain families would be of interest, and is left for future work.
Redesign of four specificity positions in Tiam1 allowed us to test the design model
in a different way. It revealed some of the limitations of fixed backbone design, but also
gave semi-quantitative agreement with available binding free energies. This agreement
predicts new mutations that could be of interest for obtaining new specificity properties.
They remain to be studied further and tested experimentally. Here, the apo and two holo
states were studied, and designed separately. Information about binding affinities and
30

binding specificity were then obtained by comparing the energies sampled in the different
simulations. In the future, we would like to include binding affinity and/or specificity
directly in the design calculations, as a property to be designed for or against within a
single simulation. In addition, we should allow different backbone structures for the apo
and each holo system. This could be done in the future, since recent hybrid Monte Carlo
schemes 35,81 can be used for multi-backbone design, and can be extended to the problem
of designing ligand binding specificity. We also note that since our energy function is
physics-based, it has transferability to a range of molecule types, such as nonnatural
amino acids (considered in an earlier protein-ligand design study 34 ). Such amino acids
could be of interest for designing PDZ peptide ligands, to provide additional diversity and
perhaps enhanced resistance to proteolysis.



c The

number of non-Gly, non-Pro positions, which can mutate during the design simulations.

39

Table 2: Unfolded state reference energies Etr (kcal/mol)
Design, P =8

une

Design, P =4


Energies within an extended peptide structure (averaged over positions). b His protonation states.

40

Table 3: Amino acid composition (%) of natural and designed PDZ proteins



Compositions are given for buried/exposed positions,
41for individual amino acid types (left) and for classes
(right); values in brackets (right) are the deviations between design and experiment per class. le
experimental target set included the Tiam1 and Cask homologs.

Table 4: Fold recognition of designed sequences by Superfamily


The average match length for sequences recognized by Superfamily and the total sequence length.

Average E-values for Superfamily assignments to the correct SCOP superfamily/family.



Figure 2: Alignment of natural PDZ sequences. The top four sequences were tested in
this work. The others are the first 30 sequences from the Pfam seed alignment. Fourteen
hydrophobic core positions are indicated by red arrows and the secondary structure elements are shown for reference. The Clustal color scheme is used, where conserved amino
acids are colored according to their physical chemical properties.


Figure 3: Sequence logos for the conserved hydrophobic core of designed and natural
Tiam1 and Cask sequences. “Hom.” corresponds to the homologs that make up our
target set of sequences (used for Etr optimization). “Pfam” corresponds to the Pfam
seed alignment. Proteus sequences were generated with model  P =8. The height of each
letter is proportional to the abundance of each type at the corresponding position in
the Proteus/Rosetta simulations or the natural sequences. The color of each letter is
determined by the physical chemical properties of each amino acid type.

45

Pfam

Tiam
Monoposition Proteus Hom. Rosetta
Figure 4: Sequence logos for sixteen surface positions in Tiam1. Same representation as
Figue. 3. The “mono-position” results are from a set of simulations where only one amino
acid at a time could mutate, the rest of the protein having its native sequence (see text).
Amino acid colors as in Fig. 3.

46

Figure 5: Histogram plots showing similarity scores for designed PDZ sequences. Similarity
scores for Tiam1 (left) and Cask (right), relative to the Pfam-RP55 alignment. The scores were
computed for all positions (top), 14 core positions (middle), and 16 surface positions (bottom).
Values are shown for Proteus, Rosetta, and Pfam sequences (all compared to RP55). le
similarity score of the wildtype sequence is indicated in each panel by a vertical arrow. The top
panels include results for Proteus simulations where a bias energy was included, which explicitly
favors sequences that are similar to Pfam (dotted lines, labeled “Biased Proteus”). Notice that
the designed sequences represented in each top, middle, or bottom panel were the same; seulement
the positions included in the similarity score calculation differ between panels.

47

Figure 6: MD simulations of Tiam1 variants designed with Proteus. A) Sequences of the
wildtype (WT) PDZ domain and the ten designed variants simulated by MD. Asterisks indicate
peptide binding residues held fixed during the design simulations. B) Backbone rms deviations
over the course of an MD simulation for WT, QM, and 6 designed variants relative to the
corresponding mean MD structure. C) Mean MD structures of WT, QM, seq-6, seq-2.

48

Figure 7: A) Tiam1 sequences designed with different levels of hydrophobic bias δ, increasing
from top to bottom. The middle sequences were obtained from Proteus simulations without any
bias (δ = 0). For each bias level, five low energy sequences are shown. Hydrophobic positions
are colored according to the simulation where they appear first: from brick red (top) to light
green (bottom). The 14 hydrophobic core positions are indicated by red arrows. B) 3D Tiam1
structure (stereo) with colors as in A).

49

Figure 8: Design of four Tiam1 specificity positions. A) X-ray structures (stereo) with the
wildtype sequence (LKLL; labelled WT; PDB code 4GVD) or the quadruple mutant sequence
(MEFV; labelled QM; PDB code 4NXQ), with bound Sdc1 and Caspr4, respectively. le
four designed sidechains are shown and labelled (both WT and QM mutant types). B) Logo
representation of designed sequences with no ligand (apo) or Sdc1 or Caspr4, using the wildtype
(left) or quadruple mutant (right) X-ray structure. C) Covariance plots for the 911-912 pair:
populations of each pair of types are shown as levels of color, with yellow the most highlypopulated (5%) and red the lowest (0%). Results correspond to design simulations with the
wildtype backbone.


</pre></body></html>
