<!DOCTYPE html
  PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html
      xmlns="http://www.w3.org/1999/xhtml"
      xml:lang="en"
      lang="en">
   <head>
      <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
      <title>A new framework for computational protein design through cost function network optimization </title>
      <meta name="googlebot" content="NOODP" />
      <meta name="HW.ad-path" content="/cgi/content/full/29/17/2129" /><script type="text/javascript"
              src="http://www.deepdyve.com.gate1.inist.fr/assets/scripts/RentalLink.js"></script><script type="text/javascript">
        deepdyve.host = 'www.deepdyve.com';
        // replace with an affiliateId that we provide
	deepdyve.affiliateId = "highwire-oupjournals";
	
        // the div to fill in with the rental link
        deepdyve.divIdList = ['rentalLink', 'rentalLink2']; 
        // this is the anchor text for the rental link
	deepdyve.rentText = "Rent Article at DeepDyve";
        deepdyve.divIdMap = {'rentalLink':'Rent Article at DeepDyve','rentalLink2':'Learn more here.'};

        // document identifier and identifier type
        deepdyve.fieldName = 'journal_doi';
	deepdyve.docId = "10.1093/bioinformatics/btt374";
	</script><meta content="/bioinfo/29/17/2129.atom" name="HW.identifier" />
      <meta name="DC.Format" content="text/html" />
      <meta name="DC.Language" content="en" />
      <meta content="A new framework for computational protein design through cost function network optimization"
            name="DC.Title" />
      <meta content="10.1093/bioinformatics/btt374" name="DC.Identifier" />
      <meta content="2013-09-01" name="DC.Date" />
      <meta content="Oxford University Press" name="DC.Publisher" />
      <meta content="Seydou Traoré" name="DC.Contributor" />
      <meta content="David Allouche" name="DC.Contributor" />
      <meta content="Isabelle André" name="DC.Contributor" />
      <meta content="Simon de Givry" name="DC.Contributor" />
      <meta content="George Katsirelos" name="DC.Contributor" />
      <meta content="Thomas Schiex" name="DC.Contributor" />
      <meta content="Sophie Barbe" name="DC.Contributor" />
      <meta content="Bioinformatics" name="citation_journal_title" />
      <meta content="Bioinformatics" name="citation_journal_abbrev" />
      <meta content="1367-4803" name="citation_issn" />
      <meta content="1460-2059" name="citation_issn" />
      <meta name="citation_author" content="Seydou Traoré" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author" content="David Allouche" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author" content="Isabelle André" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author" content="Simon de Givry" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author" content="George Katsirelos" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author" content="Thomas Schiex" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author" content="Sophie Barbe" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta name="citation_author_institution"
            content="1Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, 2INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, 3CNRS, UMR5504, F-31400 Toulouse, France and 4Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France" />
      <meta content="A new framework for computational protein design through cost function network optimization"
            name="citation_title" />
      <meta content="09/01/2013" name="citation_date" />
      <meta content="29" name="citation_volume" />
      <meta content="17" name="citation_issue" />
      <meta content="2129" name="citation_firstpage" />
      <meta content="2136" name="citation_lastpage" />
      <meta content="29/17/2129" name="citation_id" />
      <meta content="29/17/2129" name="citation_id_from_sass_path" />
      <meta content="bioinfo;29/17/2129" name="citation_mjid" />
      <meta content="10.1093/bioinformatics/btt374" name="citation_doi" />
      <meta content="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129.abstract"
            name="citation_abstract_html_url" />
      <meta content="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129.full"
            name="citation_fulltext_html_url" />
      <meta content="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129.full.pdf"
            name="citation_pdf_url" />
      <meta content="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129"
            name="citation_public_url" />
      <meta content="23842814" name="citation_pmid" />
      <meta name="citation_section" content="Original Paper" />
      <meta name="robots" content="noarchive,nofollow" />
      <meta name="googlebot" content="noarchive" />
      <link href="/content/29/17/2122.short" rel="prev" />
      <link href="/content/29/17/2137.short" rel="next" />
      <link rel="stylesheet" type="text/css" media="all" href="/shared/css/hw-global.css" />
      <link rel="stylesheet" type="text/css" media="print" href="/shared/css/hw-print.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/local/css/hw-local-global.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/publisher/css/hw-publisher-global.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/publisher/css/hw-publisher-ac.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/shared/css/hw-page-content.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/shared/css/jquery.fancybox-1.3.4.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/shared/css/hw-global-colexpand.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/publisher/css/hw-publisher-page-content.css" /><script type="text/javascript" id="session-d118659645e1">var gSmudge='864457710';</script><script type="text/javascript" id="session-d118659645e3">var callbackToken='50946CDA1431EE8';</script><script type="text/javascript" id="session-d118659645e5">
                      var subCode='oupjournal_sub';
                    </script><script type="text/javascript">
            var gAuthTimeStamp = '2014-02-28T05:27:13.908-08:00';
            var gSessionId = 'ZOszIleAgYk8zRyR3qYurQ';
            var gAuthzRequired = 'true';
            var gAuthnMethods1 = 'ip';
            var gAuthnMethods2 = 'ip';
            var gAuthnIPs = '193.54.110.32';
            var gAuthnIndividuals = '';
            var gAuthnInstitutions = '81074203';
            </script><script type="text/javascript" src="/shared/js/jquery-min.js"></script><script type="text/javascript" src="/shared/js/fingerprint.js"></script><script type="text/javascript" src="/shared/js/hw-shared.js"></script><script type="text/javascript" src="/publisher/js/hw-publisher-site.js"></script><script type="text/javascript" src="/local/js/local-js-vars.js"></script><script type="text/javascript" src="/shared/js/pages/hw-content.js"></script><script type="text/javascript" src="/shared/js/util/AccessCheckLib.js"></script><script type="text/javascript" src="/shared/js/util/executeAccessCheck.js"></script><script type="text/javascript" src="/shared/js/fancybox/jquery.fancybox-1.3.4.js"></script><script type="text/javascript" src="/shared/js/fancybox/jquery.easing-1.3.pack.js"></script><script type="text/javascript"
              src="/shared/js/fancybox/jquery.mousewheel-3.0.4.pack.js"></script><script type="text/javascript" src="/shared/js/util/content.jquery.addVariantLink.js"></script><script type="text/javascript" src="/shared/js/util/hw-col-expand.js"></script><script type="text/javascript" src="/publisher/js/hw-publisher-modalwin-vars.js"></script><script type="text/javascript"
              src="/publisher/js/hw-publisher-article-dynamic-elements.js"></script><script type="text/javascript" src="/publisher/js/hw-publisher-content.js"></script><script type="text/javascript" src="/shared/js/util/hw-mathjax.js"></script><script type="text/x-mathjax-config">
            MathJax.Hub.Config({
                tex2jax: {
                  inlineMath: [["$","$"],["\\(","\\)"]],
                  skipTags: ["body"],
                  processClass: "mml-math"
                }
            });
        </script><script type="text/javascript"
              src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><script type="text/javascript">
	    
	    
	    
	    var siqDOI = encodeURIComponent("10.1093/bioinformatics/btt374");
	    var siqIsOpenAccess = encodeURIComponent("");
	    var siqPubDate = encodeURIComponent("20130901");

	    if (siqDOI.length == 0) {
	        siqDOI = "UNKNOWN";
	    }
	    
	    
		
		
	    if (gAuthnIndividuals.length != 0) {
		if (gAuthnInstitutions.length != 0) {
		    authnEntity = encodeURIComponent(gAuthnIndividuals
		        + ',' + gAuthnInstitutions);
		} else {
		    authnEntity = encodeURIComponent(gAuthnIndividuals);
		}
	    } else {
	        authnEntity = encodeURIComponent(gAuthnInstitutions);
	    }

	    var commonString =
		'authSessionId=' + gSessionId + String.fromCharCode(0x26)
		+ 'authzRequired=' + gAuthzRequired + String.fromCharCode(0x26)
		+ 'authentication_method=' + encodeURIComponent(gAuthnMethods2) + String.fromCharCode(0x26)
		+ 'authnIPs=' + gAuthnIPs + String.fromCharCode(0x26)
		+ 'authnInstitutions=' + authnEntity;

		
		
		
		

	    var gPageId = "pageid-content";
	    var gVariant = "full-text";

	    
		// Not completely done
		var eventType = "full-text";
		var accessType;
		if (siqIsOpenAccess == 'true') {
		    accessType = 'SOA';
		} else {
		    accessType = 'subscription';
		}
		    
		var NTPT_PGEXTRA =
		    commonString + String.fromCharCode(0x26)
		    + 'event_type=' + eventType + String.fromCharCode(0x26)
		    + 'publication_date=' + siqPubDate + String.fromCharCode(0x26)
		    + 'access_type=' + accessType + String.fromCharCode(0x26)
		    + 'doi=' + siqDOI ;
		    
		

	    // alert("NTPT_PGEXTRA is " + NTPT_PGEXTRA);
	</script><link rel="stylesheet" type="text/css" media="all" href="/resource/css/hw20.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/resource/css/h20-oup-header.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/resource/css/h20-oup-footer.css" />
      <link rel="stylesheet" type="text/css" media="all"
            href="/resource/css/h20-oup-sidebars.css" />
      <link type="text/css" media="all" rel="stylesheet"
            href="/site/resource/bioinformatics_h20.css" />
      <link rel="stylesheet" type="text/css" href="/resource/css/print.css" media="print" /><script type="text/javascript" src="/resource/js/main.js"></script><link rel="stylesheet" type="text/css" media="all"
            href="http://oi-underbar.ifactory.com/underbar/css/pf_oiunderbar.css" />
      <link rel="stylesheet" type="text/css" media="screen"
            href="http://gab.cookie.oup.com.gate1.inist.fr/aws-cookie/jquery.fancybox-1.3.4_1.css" /><script type="text/javascript" src="/resource/js/oup_ad_size.js"></script></head>
   <body class="general_page journal bioinformatics">
      <div class="hw-gen-page pagetype-content hw-pub-id-article" id="pageid-content"
           itemscope="itemscope"
           itemtype="http://schema.org/ScholarlyArticle">
         <noscript>
            <p>We use cookies to enhance your experience on our website. By continuing to use our website, you are agreeing to our use of
               cookies. You can change your cookie settings at any time. <a href="http://global.oup.com.gate1.inist.fr/cookiepolicy/">Find out more</a></p>
         </noscript>
         
         
         <p class="hide">
            <a href="#content">Skip Navigation</a>
            
         </p>
         
         
         <div id="secondary_nav">
            <strong id="page_logo" title="Oxford Journals"><a href="http://www.oxfordjournals.org.gate1.inist.fr/"><span>Oxford Journals</span></a></strong>
            
            <ul>
               <li id="nav_contact_us" title="Contact Us"><a href="http://www.oxfordjournals.org.gate1.inist.fr/contact_us.html"><span>Contact Us</span></a></li>
               <li id="nav_my_basket" title="My Basket"><a href="https://secure-oxfordjournals-org.gate1.inist.fr/basket.html"><span>My Basket</span></a></li>
               <li id="nav_my_account" title="My Account"><a href="http://services.oxfordjournals.org.gate1.inist.fr/cgi/tslogin?url=http://www.oxfordjournals.org.gate1.inist.fr/service/Register"><span>My 
                        
                        Account</span></a></li>
            </ul>
            
         </div>
         
         
         <div id="header">
            
            <h1 id="page_title" title="Bioinformatics"><a href="/"><span>
                     Bioinformatics
                     </span></a></h1>
            
         </div>
         
         
         <div id="primary_nav">
            
            <ul>
               <li id="nav_about_this_journal" title="About This Journal">
                  <a href="http://www.oxfordjournals.org.gate1.inist.fr/bioinformatics/about.html">
                     
                     <span>About This Journal</span>
                     </a>
                  
               </li>
               <li id="nav_contact_this_journal" title="Contact This Journal"><a href="/cgi/feedback/"><span>Contact This Journal</span></a></li>
               <li id="nav_subscriptions" title="Subscriptions">
                  <a href="http://www.oxfordjournals.org.gate1.inist.fr/bioinformatics/access_purchase/buy_online.html">
                     
                     <span>Subscriptions</span>
                     </a>
                  
               </li>
               <li id="nav_current_issue" title="Current"><a href="/content/current"><span>View Current Issue (Volume 30 Issue 5 March 1, 2014)</span></a></li>
               <li id="nav_archive" title="Archive"><a href="/content/by/year"><span>Archive</span></a></li>
               <li id="nav_search" title="Search"><a href="/search"><span>Search</span></a></li>
            </ul>
            
         </div>
         
         
         <div id="user_nav">
            
            <div class="header-ac-elements">
               
               <div id="authstring">
                  
                  <ul>
                     <li class="subscr-ref">Institution:  INIST-CNRS</li>
                     <li class="no-left-border">
                        <a href="/login?uri=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcontent%2F29%2F17%2F2129.full">
                           Sign In as Personal Subscriber
                           </a>
                        
                     </li>
                  </ul>           
                  
               </div>   
               
            </div>
            
         </div>
         
         
         <ul id="site-breadcrumbs">
            <li class="first"><a href="http://services.oxfordjournals.org.gate1.inist.fr/cgi/tslogin?url=http%3A%2F%2Fwww.oxfordjournals.org">Oxford Journals</a></li>
            <li><span class="breadcrumb_subjects"><a href="http://www.oxfordjournals.org.gate1.inist.fr/subject/life_sciences/"
                     class="breadcrumb_subject">Life Sciences</a></span><span class="breadcrumb_subjects"><a href="http://www.oxfordjournals.org.gate1.inist.fr/subject/mathematics/"
                     class="breadcrumb_subject">Mathematics &amp; Physical Sciences</a></span></li>
            <li><a href="/">Bioinformatics</a></li>
            <li><a href="/content/29/17.toc">
                  <span xmlns="" class="volume-value">Volume 29</span>
                  <span xmlns="" class="issue-value"> Issue 17</span></a></li>
            <li>Pp. <span class="slug-pages">
                  2129-2136.
                  </span></li>
         </ul>
         
         
         
         
         <div id="oas_top" class="ad_hidden">
            <iframe id="id_advertframe_top"
                    src="/resource/htmlfiles/advert.html?p=Top&amp;u=bioinformatics.oxfordjournals.org/content/29/17/2129.full"
                    width="0"
                    height="0"
                    frameborder="0"
                    scrolling="no"
                    class="resize_ad"></iframe>
            
         </div>
         
         
         <a name="content"></a>
         
         
         <div id="h20_page"></div>
         
         <div id="content-block">
            <div class="article fulltext-view " itemprop="articleBody"><span class="highwire-journal-article-marker-start"></span><h1 id="article-title-1" itemprop="headline">A new framework for computational protein design through cost function network optimization</h1>
               <div class="contributors">
                  <ol class="contributor-list" id="contrib-group-1">
                     <li class="contributor" id="contrib-1" itemprop="author" itemscope="itemscope"
                         itemtype="http://schema.org/Person"><span class="name" itemprop="name"><a class="name-search"
                              href="/search?author1=Seydou+Traor%C3%A9&amp;sortspec=date&amp;submit=Submit">Seydou Traoré</a></span><a id="xref-aff-1-1" class="xref-aff" href="#aff-1"><sup>1</sup></a><span class="xref-sep">,</span><a id="xref-aff-1-2" class="xref-aff" href="#aff-1"><sup>2</sup></a><span class="xref-sep">,</span><a id="xref-aff-1-3" class="xref-aff" href="#aff-1"><sup>3</sup></a><span class="xref-sep">,</span><a id="xref-fn-1-1" class="xref-fn" href="#fn-1"><sup>†</sup></a>, 
                     </li>
                     <li class="contributor" id="contrib-2" itemprop="author" itemscope="itemscope"
                         itemtype="http://schema.org/Person"><span class="name" itemprop="name"><a class="name-search"
                              href="/search?author1=David+Allouche&amp;sortspec=date&amp;submit=Submit">David Allouche</a></span><a id="xref-aff-1-4" class="xref-aff" href="#aff-1"><sup>4</sup></a><span class="xref-sep">,</span><a id="xref-fn-1-2" class="xref-fn" href="#fn-1"><sup>†</sup></a>, 
                     </li>
                     <li class="contributor" id="contrib-3" itemprop="author" itemscope="itemscope"
                         itemtype="http://schema.org/Person"><span class="name" itemprop="name"><a class="name-search"
                              href="/search?author1=Isabelle+Andr%C3%A9&amp;sortspec=date&amp;submit=Submit">Isabelle André</a></span><a id="xref-aff-1-5" class="xref-aff" href="#aff-1"><sup>1</sup></a><span class="xref-sep">,</span><a id="xref-aff-1-6" class="xref-aff" href="#aff-1"><sup>2</sup></a><span class="xref-sep">,</span><a id="xref-aff-1-7" class="xref-aff" href="#aff-1"><sup>3</sup></a>, 
                     </li>
                     <li class="contributor" id="contrib-4" itemprop="author" itemscope="itemscope"
                         itemtype="http://schema.org/Person"><span class="name" itemprop="name"><a class="name-search"
                              href="/search?author1=Simon+de+Givry&amp;sortspec=date&amp;submit=Submit">Simon de Givry</a></span><a id="xref-aff-1-8" class="xref-aff" href="#aff-1"><sup>4</sup></a>, 
                     </li>
                     <li class="contributor" id="contrib-5" itemprop="author" itemscope="itemscope"
                         itemtype="http://schema.org/Person"><span class="name" itemprop="name"><a class="name-search"
                              href="/search?author1=George+Katsirelos&amp;sortspec=date&amp;submit=Submit">George Katsirelos</a></span><a id="xref-aff-1-9" class="xref-aff" href="#aff-1"><sup>4</sup></a>, 
                     </li>
                     <li class="contributor" id="contrib-6" itemprop="author" itemscope="itemscope"
                         itemtype="http://schema.org/Person"><span class="name" itemprop="name"><a class="name-search"
                              href="/search?author1=Thomas+Schiex&amp;sortspec=date&amp;submit=Submit">Thomas Schiex</a></span><a id="xref-aff-1-10" class="xref-aff" href="#aff-1"><sup>4</sup></a><span class="xref-sep">,</span><a id="xref-corresp-1-1" class="xref-corresp" href="#corresp-1">*</a> and 
                     </li>
                     <li class="last" id="contrib-7"><span class="name"><a class="name-search"
                              href="/search?author1=Sophie+Barbe&amp;sortspec=date&amp;submit=Submit">Sophie Barbe</a></span><a id="xref-aff-1-11" class="xref-aff" href="#aff-1"><sup>1</sup></a><span class="xref-sep">,</span><a id="xref-aff-1-12" class="xref-aff" href="#aff-1"><sup>2</sup></a><span class="xref-sep">,</span><a id="xref-aff-1-13" class="xref-aff" href="#aff-1"><sup>3</sup></a><span class="xref-sep">,</span><a id="xref-corresp-1-2" class="xref-corresp" href="#corresp-1">*</a></li>
                  </ol>
                  <ol class="affiliation-list">
                     <li class="aff"><a id="aff-1" name="aff-1"></a><address><sup>1</sup>Université de Toulouse; INSA, UPS, INP; LISBP, 135 Avenue de Rangueil, F-31077 Toulouse, France, <sup>2</sup>INRA, UMR792, Ingénierie des Systèmes Biologiques et des Procédés, F-31400 Toulouse, France, <sup>3</sup>CNRS, UMR5504, F-31400 Toulouse, France and <sup>4</sup>Unité de Mathématiques et Informatique Appliquées, UR 875, INRA, F-31320 Castanet Tolosan, France
                        </address>
                     </li>
                  </ol>
                  <ol class="corresp-list">
                     <li class="corresp" id="corresp-1"><a class="rev-xref" href="#xref-corresp-1-1">↵</a>*To whom correspondence should be addressed.
                     </li>
                  </ol>
                  <ul class="history-list">
                     <li xmlns:hwp="http://schema.highwire.org.gate1.inist.fr/Journal" class="received"
                         hwp:start="2013-03-07"><span class="received-label">Received </span>March 7, 2013.
                     </li>
                     <li xmlns:hwp="http://schema.highwire.org.gate1.inist.fr/Journal" class="rev-recd"
                         hwp:start="2013-05-29"><span class="rev-recd-label">Revision received </span>May 29, 2013.
                     </li>
                     <li xmlns:hwp="http://schema.highwire.org.gate1.inist.fr/Journal" class="accepted"
                         hwp:start="2013-06-21"><span class="accepted-label">Accepted </span>June 21, 2013.
                     </li>
                  </ul>
               </div>
               <div class="section abstract" id="abstract-1" itemprop="description">
                  <div class="section-nav">
                     <div class="nav-placeholder"> </div><a href="#sec-1" title="1 INTRODUCTION" class="next-section-link"><span>Next Section</span></a></div>
                  <h2>Abstract</h2>
                  
                  <p id="p-3"><strong>Motivation:</strong> The main challenge for structure-based computational protein design (CPD) remains the combinatorial nature of the search
                     space. Even in its simplest fixed-backbone formulation, CPD encompasses a computationally difficult NP-hard problem that prevents
                     the exact exploration of complex systems defining large sequence-conformation spaces.
                  </p>
                  
                  <p id="p-4"><strong>Results:</strong> We present here a CPD framework, based on cost function network (CFN) solving, a recent exact combinatorial optimization
                     technique, to efficiently handle highly complex combinatorial spaces encountered in various protein design problems. We show
                     that the CFN-based approach is able to solve optimality a variety of complex designs that could often not be solved using
                     a usual CPD-dedicated tool or state-of-the-art exact operations research tools. Beyond the identification of the optimal solution,
                     the global minimum-energy conformation, the CFN-based method is also able to quickly enumerate large ensembles of suboptimal
                     solutions of interest to rationally build experimental enzyme mutant libraries.
                  </p>
                  
                  <p id="p-5"><strong>Availability:</strong> The combined pipeline used to generate energetic models (based on a patched version of the open source solver Osprey 2.0),
                     the conversion to CFN models (based on Perl scripts) and CFN solving (based on the open source solver toulbar2) are all available
                     at <a href="http://genoweb.toulouse.inra.fr/~tschiex/CPD">http://genoweb.toulouse.inra.fr/∼tschiex/CPD</a></p>
                  
                  <p id="p-6"><strong>Contacts:</strong> <span class="em-link"><span class="em-addr">thomas.schiex{at}toulouse.inra.fr</span></span> or <span class="em-link"><span class="em-addr">sophie.barbe{at}insa-toulouse.fr</span></span></p>
                  
                  <p id="p-7"><strong>Supplementary information:</strong> <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary data</a> are available at <em>Bioinformatics</em> online.
                  </p>
                  
               </div>
               <div class="section intro" id="sec-1">
                  <div class="section-nav"><a href="#abstract-1" title="Abstract" class="prev-section-link"><span>Previous Section</span></a><a href="#sec-2" title="2 METHODS" class="next-section-link"><span>Next Section</span></a></div>
                  <h2>1 INTRODUCTION</h2>
                  
                  <p id="p-8">The engineering of tailored proteins with desired properties holds great interest for applications ranging from medicine,
                     biotechnology (<a id="xref-ref-42-1" class="xref-bibr" href="#ref-42">Nestl <em>et al.</em>, 2011</a>) and synthetic biology to nanotechnologies (<a id="xref-ref-23-1" class="xref-bibr" href="#ref-23">Grunwald <em>et al.</em>, 2009</a>). Although, directed evolution techniques coupled with high-throughput automated procedures have met with some success, they
                     do not provide structural design principles to guide the rational design of novel proteins. The development of generic and
                     effective protein engineering methodologies, both experimental and computational, is thus of utmost interest to speedup the
                     design of tailored proteins having the desired properties. Structure-based computational protein design (CPD) approaches have
                     demonstrated their potential to adequately capture fundamental aspects of molecular recognition and interactions, which have
                     already enabled the successful (re)design of several enzymes for various purposes (<a id="xref-ref-12-1" class="xref-bibr" href="#ref-12">Dahiyat and Mayo, 1997</a>; <a id="xref-ref-26-1" class="xref-bibr" href="#ref-26">Hellinga and Richards, 1991</a>). Despite these outstanding results, the efficiency, predictability and reliability of CPD methods have shown that they still
                     need to mature.
                  </p>
                  
                  <p id="p-9">CPD is faced with several challenges. The first lies in the exponential size of the conformational and protein sequence space
                     that has to be explored, which rapidly grows out of reach of computational approaches. Another obstacle to overcome is the
                     unsolved issue of accurate structure prediction for a given sequence. Therefore, the design problem is usually approached
                     as an inverse folding problem (<a id="xref-ref-43-1" class="xref-bibr" href="#ref-43">Pabo, 1983</a>), to reduce the problem to the identification of an amino acid sequence that can fold into a target protein 3D-scaffold that
                     matches the design objective. This paradigm typically assumes a fixed protein backbone and, for each type of amino acid considered
                     at a given position, allows the side chains to move only among a set of discrete and low-energy conformations, called rotamers
                     (<a id="xref-ref-30-1" class="xref-bibr" href="#ref-30">Janin <em>et al.</em>, 1978</a>). CPD is thus formulated as an optimization problem, which consists in choosing combinations of rotamers at designable specified
                     positions such that the fold is stabilized and the desired property is achieved. To solve this problem, we need a computationally
                     tractable energetic model to evaluate the energy of any combination of rotamers. We also require computational optimization
                     techniques that can efficiently explore the sequence-conformation space to find the sequence-conformation model of global
                     minimum energy (GMEC: global minimum-energy conformation) or an ensemble of low-energy sequence-conformation models. Indeed,
                     several reasons can motivate the generation of multiple near-optimal solutions. First, the sequence-conformation model with
                     the lowest predicted energy may not fold into the targeted protein scaffold owing to inaccuracies in the modeling of protein
                     energetics. Secondly, the GMEC solution may be so stabilized that it can lack the flexibility required to operate the protein
                     biological function (<a id="xref-ref-5-1" class="xref-bibr" href="#ref-5">Arnold, 2001</a>). Such suboptimal ensembles can then be analyzed to rationally guide the experimental construction of protein libraries while
                     enhancing the chances of success to identify a protein hit.
                  </p>
                  
                  <p id="p-10">The protein design problem modeled with a rigid backbone, a discrete set of rotamers and pairwise energy functions, has been
                     proven to be NP-hard (<a id="xref-ref-44-1" class="xref-bibr" href="#ref-44">Pierce and Winfree, 2002</a>). Hence several meta-heuristic methods have been applied to it, including Monte-Carlo with simulated annealing (<a id="xref-ref-34-1" class="xref-bibr" href="#ref-34">Kuhlman and Baker, 2000</a>; <a id="xref-ref-48-1" class="xref-bibr" href="#ref-48">Voigt <em>et al.</em>, 2000</a>), genetic algorithms (<a id="xref-ref-15-1" class="xref-bibr" href="#ref-15">Desjarlais and Handel, 1995</a>; <a id="xref-ref-45-1" class="xref-bibr" href="#ref-45">Raha <em>et al.</em>, 2000</a>) and other methods (<a id="xref-ref-1-1" class="xref-bibr" href="#ref-1">Allen and Mayo, 2006</a>; <a id="xref-ref-17-1" class="xref-bibr" href="#ref-17">Desmet <em>et al.</em>, 2002</a>; <a id="xref-ref-52-1" class="xref-bibr" href="#ref-52">Wernisch <em>et al.</em>, 2000</a>). These approaches can usually find a relatively low-energy model fairly quickly but without any guarantees of completeness
                     or accuracy. Indeed, these stochastic optimization routines may end up trapped in local minima and miss the GMEC with no indication.
                  </p>
                  
                  <p id="p-11">Conversely, there exist methods that solve the GMEC exactly, such as approaches based on the dead-end elimination (DEE) theorem
                     (<a id="xref-ref-16-1" class="xref-bibr" href="#ref-16">Desmet <em>et al.</em>, 1992</a>), on the branch-and-bound algorithm (<a id="xref-ref-22-1" class="xref-bibr" href="#ref-22">Gordon and Mayo, 1999</a>; <a id="xref-ref-27-1" class="xref-bibr" href="#ref-27">Hong <em>et al.</em>, 2009</a>; <a id="xref-ref-52-2" class="xref-bibr" href="#ref-52">Wernisch <em>et al.</em>, 2000</a>), on integer linear programming (<a id="xref-ref-3-1" class="xref-bibr" href="#ref-3">Althaus <em>et al.</em>, 2002</a>; <a id="xref-ref-31-1" class="xref-bibr" href="#ref-31">Kingsford <em>et al.</em>, 2005</a>) or on dynamic programming (<a id="xref-ref-39-1" class="xref-bibr" href="#ref-39">Leaver-Fay <em>et al.</em>, 2005</a>). These exact methods offer several advantages. First, they ensure that discrepancies between CPD predictions and experimental
                     results come exclusively from the inadequacies of the biophysical model and not from the algorithm. Next, because provable
                     methods can determine that the optimum is reached, they may actually stop before meta-heuristic approaches. Finally, empirical
                     studies on solving the GMEC problem reported that the accuracy of meta-heuristic approaches tend to degrade as the problem
                     size increases (<a id="xref-ref-48-2" class="xref-bibr" href="#ref-48">Voigt <em>et al.</em>, 2000</a>).
                  </p>
                  
                  <p id="p-12">In this article, we modeled the CPD problem as either a binary cost function network (CFN) or an integer linear programming
                     (ILP) problem (Section 2.4). We compared the performance of the open source CFN solver <em>toulbar2</em> and the IBM™ ILOG ILP solver <em>cplex</em> against that of the combined DEE/A* approach as implemented in the dedicated CPD software <em>Osprey</em> (Section 3.2), for design problems (Section 3.1). The CFN-based method outperformed by several orders of magnitude the other
                     methods both in identifying the GMEC but also in producing a set of low-energy sequence-conformation models. This second step
                     was not attainable in most of the study cases using DEE/A* (Section 3.3). Therefore, on the basis of the CFN approach, we
                     propose a new CPD framework (<a id="xref-fig-1-1" class="xref-fig" href="#F1">Fig. 1</a>). Our methodology, which we describe in Section 2, is well-adapted to solving exactly macromolecular design problems of large
                     sequence-conformation spaces. It also has the potential to improve methods that integrate flexibility to a larger extent in
                     protein design, as this considerably expands the size of the search space or may require solving a large number of GMEC instances
                     (<a id="xref-ref-24-1" class="xref-bibr" href="#ref-24">Hallen <em>et al.</em>, 2013</a>; <a id="xref-ref-29-1" class="xref-bibr" href="#ref-29">Humphris and Kortemme, 2008</a>). These aspects are highly relevant to CPD and we shall address them here.
                     
                  </p>
                  <div id="F1" class="fig pos-float  odd">
                     <div class="fig-inline"><a href="2129/F1.expansion.html"><img alt="Fig. 1." src="2129/F1.small.gif" /></a><div class="callout"><span>View larger version:</span><ul class="callout-links">
                              <li><a href="2129/F1.expansion.html">In this window</a></li>
                              <li><a class="in-nw" href="2129/F1.expansion.html">In a new window</a></li>
                           </ul>
                           <ul class="fig-services">
                              <li class="ppt-link"><a href="/powerpoint/29/17/2129/F1">Download as PowerPoint Slide</a></li>
                           </ul>
                        </div>
                     </div>
                     <div class="fig-caption"><span class="fig-label">Fig. 1.</span> 
                        <p id="p-13" class="first-child">Flow chart of the CPD framework</p>
                        <div class="sb-div caption-clear"></div>
                     </div>
                  </div>
               </div>
               <div class="section methods" id="sec-2">
                  <div class="section-nav"><a href="#sec-1" title="1 INTRODUCTION" class="prev-section-link"><span>Previous Section</span></a><a href="#sec-11" title="3 RESULTS AND DISCUSSION" class="next-section-link"><span>Next Section</span></a></div>
                  <h2>2 METHODS</h2>
                  
                  <p id="p-14">The CPD strategy introduced in this work (<a id="xref-fig-1-2" class="xref-fig" href="#F1">Fig. 1</a>) is composed of five main stages discussed in more details in the following subsections. The whole CPD framework and the
                     approaches used to handle the sequence-conformation combinatorial optimization problem were assessed for the design of more
                     stable proteins and cofactor-bound proteins as well as protein–ligand and protein–protein interfaces.
                  </p>
                  
                  <div id="sec-3" class="subsection">
                     <h3>2.1 Preparation of structural molecular systems</h3>
                     
                     <p id="p-15">Three-dimensional models of proteins in free and complex states were derived from their respective crystallographic structures
                        deposited in the protein data bank (PDB) (<a id="xref-ref-6-1" class="xref-bibr" href="#ref-6">Bernstein <em>et al.</em>, 1977</a>) (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>). Missing heavy atoms in crystal structures as well as hydrogen atoms were added using the <em>tleap</em> module of the <em>Amber 9</em> software package (<a id="xref-ref-9-1" class="xref-bibr" href="#ref-9">Case <em>et al.</em>, 2006</a>). Cofactors as well as crystallographic water molecules specified in <em>SITE</em> and <em>LINK</em> records of PDB files were kept in structural models. Histidine protonation states and disulfide bonds were assigned using
                        the <em>tleap</em> module. For multimeric proteins, the transformation matrix specified in the PDB file was applied to reproduce missing chains.
                        Parameters for non-amino acid type ligands and cofactors were generated with the <em>Antechamber</em> module of <em>Amber 9</em> (<a id="xref-ref-51-1" class="xref-bibr" href="#ref-51">Wang <em>et al.</em>, 2006</a>). The molecular all-atom <em>ff99SB</em> (<a id="xref-ref-28-1" class="xref-bibr" href="#ref-28">Hornak <em>et al.</em>, 2006</a>), <em>Glycam06</em> (<a id="xref-ref-32-1" class="xref-bibr" href="#ref-32">Kirschner <em>et al.</em>, 2008</a>) and <em>gaff</em> force fields (<a id="xref-ref-50-1" class="xref-bibr" href="#ref-50">Wang <em>et al.</em>, 2004</a>) were used for the proteins, carbohydrates and other non-standard molecules, respectively. Each molecular system was then
                        subjected to 500 steps of minimization with the <em>Sander</em> module of <em>Amber 9</em>, using the generalized born/surface area implicit solvent model (<a id="xref-ref-25-1" class="xref-bibr" href="#ref-25">Hawkins <em>et al.</em>, 1996</a>).
                     </p>
                  </div>
                  
                  <div id="sec-4" class="subsection">
                     <h3>2.2 Definition of sequence-conformation spaces</h3>
                     
                     <p id="p-16">The residues of each protein were classified into three layers (labeled core, boundary or surface) according to their burial
                        in the 3D-model (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Fig. S1</a>). This burial of residues was defined by calculating their solvation radius from atomic solvation radii, as defined by (<a id="xref-ref-4-1" class="xref-bibr" href="#ref-4">Archontis and Simonson, 2005</a>). The salvation radius B<sub>R</sub> of residue R is given as follows:
                        <span class="disp-formula" id="disp-formula-1"><img class="graphic" alt="Formula" src="2129/embed/graphic-2.gif" /><span class="disp-formula-label">(1)</span></span>
                        where b<sub>i</sub> and q<sub>i</sub> are the atomic solvation radius of atom <em>i</em> from residue <em>R</em> and its partial charge, respectively. From these calculations, three layers of decreasing residue-solvation radius from the
                        core to the surface of the protein were defined. The set of amino acid types considered at each mutable residue (i.e. candidate
                        positions for redesign) of the proteins (in their apo form or bound to a cofactor) depends on the layer to which the residue
                        belongs to. Further details regarding the selection of designed positions and the allowed amino acids can be found in the
                        <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary data</a>.
                     </p>
                  </div>
                  
                  <div id="sec-5" class="subsection">
                     <h3>2.3 Computation of pairwise energies</h3>
                     
                     <p id="p-17">The total energy (E<sub>total</sub>) of a sequence-conformation model defined by the selection of one specific amino acid associated with a given conformation
                        (rotamer) for each variable amino acid type is assessed as follows:
                        <span class="disp-formula" id="disp-formula-2"><img class="graphic" alt="Formula" src="2129/embed/graphic-3.gif" /><span class="disp-formula-label">(2)</span></span>
                        where <em>E</em><sub>c</sub> is a constant energy contribution capturing interactions between fixed parts of the model, <span class="inline-formula" id="inline-formula-1"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-1.gif" /></span> is the difference between the self energy of rotamer <em>r</em> at position <em>i</em> capturing internal interactions or interactions with fixed regions <span class="inline-formula" id="inline-formula-2"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-2.gif" /></span> and its reference energy <span class="inline-formula" id="inline-formula-3"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-3.gif" /></span>, which corresponds to the lowest computed intra-rotamer energy for each amino acid type by variable residue position (<a id="xref-ref-41-1" class="xref-bibr" href="#ref-41">Lippow <em>et al.</em>, 2007</a>) and <span class="inline-formula" id="inline-formula-4"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-4.gif" /></span> is the pairwise interaction energy between rotamer <em>r</em> at position <em>i</em> and rotamer <em>s</em> at position <em>j</em>. In this formulation, the conformations (i.e. rotamers) of amino acid–type ligands are processed as rotamers of amino acid
                        side chains.
                     </p>
                     
                     <p id="p-18">All pairwise energy terms were pre-computed and stored using <em>Osprey 2.0</em> (<a id="xref-ref-18-1" class="xref-bibr" href="#ref-18">Gainza <em>et al.</em>, 2012</a>). These calculations were based on the <em>Amber</em> all-atom <em>ff94</em> parameters implemented in <em>Osprey 2.0</em> as well as on additional force field parameters generated from <em>Glycam06</em> and <em>gaff</em> force fields using the <em>Antechamber</em> module of <em>Amber 9</em>. These parameters were added in the parameter files of <em>Osprey 2.0</em> (<a id="xref-ref-10-1" class="xref-bibr" href="#ref-10">Chen <em>et al.</em>, 2009</a>) and were used for modeling carbohydrates and other non-standard molecules. The energy functions consisted in the sum of
                        the <em>Amber</em> electrostatic terms (with a distance-dependent dielectric constant), van der Waals and dihedral energy terms and the EEF1
                        implicit solvation energy term. No cutoff was used for non-bonded interactions.
                     </p>
                  </div>
                  
                  <div id="sec-6" class="subsection">
                     <h3>2.4 Sequence-conformation optimization</h3>
                     
                     <p id="p-19">The problem of finding the set of rotamers that will optimize the total energy (<span class="inline-formula" id="inline-formula-5"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-5.gif" /></span>) was modeled as either a binary CFN or an ILP problem. The complete interaction graph and large tree-width excluded the use
                        of dynamic programming (<a id="xref-ref-39-2" class="xref-bibr" href="#ref-39">Leaver-Fay <em>et al.</em>, 2005</a>). The performance of CFN and ILP was compared against that of the combined DEE/A* CPD-dedicated approach.
                     </p>
                     
                     <div id="sec-7" class="subsection">
                        <h4>CFN model</h4>
                        
                        <p id="p-20">A <em>CFN</em>, or weighted constraint satisfaction problem P, is composed of a set of local cost functions, each involving a set of specific
                           variables (<a id="xref-ref-47-1" class="xref-bibr" href="#ref-47">Schiex <em>et al.</em>, 1995</a>). CFNs have been used as a modeling framework for representing and solving various combinatorial optimization problems in
                           bioinformatics and resource allocation (<a id="xref-ref-8-1" class="xref-bibr" href="#ref-8">Cabon <em>et al.</em>, 1999</a>; <a id="xref-ref-14-1" class="xref-bibr" href="#ref-14">De Givry <em>et al.</em>, 2006</a>; <a id="xref-ref-53-1" class="xref-bibr" href="#ref-53">Zytnicki <em>et al.</em>, 2008</a>).
                        </p>
                        
                        <p id="p-21">Formally, a CFN P is a triple <span class="inline-formula" id="inline-formula-6"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-6.gif" /></span>, where X = {1, 2, … , n} is a set of <em>n</em> variables. Each variable <span class="inline-formula" id="inline-formula-7"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-7.gif" /></span> has a discrete domain <span class="inline-formula" id="inline-formula-8"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-8.gif" /></span>. <em>C</em> is a set of local cost functions. Each cost function <span class="inline-formula" id="inline-formula-9"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-9.gif" /></span> is defined over a subset of variables <span class="inline-formula" id="inline-formula-10"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-10.gif" /></span> (called its scope), has domain <span class="inline-formula" id="inline-formula-11"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-11.gif" /></span> and takes its values in <span class="inline-formula" id="inline-formula-12"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-12.gif" /></span>. Cost functions must be non-negative but are otherwise totally arbitrary and are often described by cost tables. The infinite
                           cost is used to represent hard constraints. An assignment <em>A</em> is a mapping from variables to values from their domains. The cost of an assignment <em>A</em> for a local cost function is the value of the cost function for the projection of <em>A</em> to the scope of the function. The global cost of <em>A</em> is the sum of the costs of <em>A</em> over all local cost functions. It is usually assumed that <em>C</em> contains one constant cost function, with an empty scope, denoted as <span class="inline-formula" id="inline-formula-13"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-13.gif" /></span>. A CFN <em>P</em> defines a joint cost distribution over all the variables <em>X</em> defined by the cost of the assignments. Because all cost functions in a CFN are non-negative, the constant cost function
                           <span class="inline-formula" id="inline-formula-14"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-14.gif" /></span> defines a lower bound on this joint cost distribution. Solving a CFN consists in finding an assignment that minimizes the
                           joint cost distribution.
                        </p>
                        
                        <p id="p-22">Modeling a CPD problem as a CFN is straightforward. Each mutable or flexible residue <em>i</em> is represented by a variable <em>i</em>. The set of rotamers available to the residue defines the domain <em>d<sub>i</sub></em> of the variable <em>i</em>. Finally, each interaction energy term in <span class="inline-formula" id="inline-formula-15"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-15.gif" /></span> can be represented as a cost function. The constant term <em>E<sub>c</sub></em> is captured as a constant cost function with empty scope. The terms <span class="inline-formula" id="inline-formula-16"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-16.gif" /></span>, which depend on one residue only, are captured as unary cost functions involving one variable <em>i</em> each. Finally, interaction terms <span class="inline-formula" id="inline-formula-17"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-17.gif" /></span> can be captured as binary cost functions involving variables <em>i</em> and <em>j</em>. To enforce the non-negativity requirement on cost functions, one can simply subtract the minimum of every cost function
                           from its cost table. The joint cost distribution defined by the corresponding CFN is then equal to the energy, up to a known
                           constant shift. The optimal solution of the CFN is an assignment that corresponds to a GMEC for the CPD problem. When the
                           maximum number of available rotamers over all residues is <em>d</em>, the resulting binary CFN takes space in <span class="inline-formula" id="inline-formula-18"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-18.gif" /></span>.
                        </p>
                        
                        <p id="p-23">Solving a CFN consists in finding a combination of values for all the variables in <em>X</em> that minimizes the joint cost distribution. Such an optimal solution defines a GMEC for the CPD problem. Existing exact algorithms
                           for solving CFN are usually depth-first branch and bound (DFBB) algorithms integrating strong incrementally computed polynomial
                           time lower bounds. The use of depth-first search algorithms avoids the worst-case exponential space of the A* algorithm used
                           in the DEE/A* approach. The incremental lower bounds are produced by algorithms that enforce so-called local consistencies
                           (<a id="xref-ref-35-1" class="xref-bibr" href="#ref-35">Larrosa and Schiex, 2004</a>; <a id="xref-ref-46-1" class="xref-bibr" href="#ref-46">Schiex, 2000</a>). Enforcing a local consistency on a given CFN <em>P</em> transforms it to an <em>equivalent</em> CFN <em>P</em>’ (defining the same joint cost distribution) with a possibly increased constant cost function <span class="inline-formula" id="inline-formula-19"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-19.gif" /></span>, providing an improved lower bound. This lower bound is used to prune the search tree and to delete rotamers.
                        </p>
                        
                        <p id="p-24">To identify the GMEC of all CPD instances, we used the open source <em>toulbar2</em> (Toulbar2 is an international collaborative CFN solver development. All sources are available on our software forge at <a href="http://mulcyber.toulouse.inra.fr/projects/toulbar2">http://mulcyber.toulouse.inra.fr/projects/toulbar2</a>) solver (release 0.9.6.0) with options −d:−l = 3 -m and no initial upper bound. By default, <em>toulbar2</em> maintains existential directional arc consistency (<a id="xref-ref-13-1" class="xref-bibr" href="#ref-13">De Givry <em>et al.</em>, 2005</a>) for incremental lower bounding, dynamic value ordering (based on minimum unary cost) and a variable ordering heuristics
                           (based on the median energy of terms involving a given residue following preprocessing) combined with last conflict heuristics
                           (<a id="xref-ref-40-1" class="xref-bibr" href="#ref-40">Lecoutre <em>et al.</em>, 2009</a>). The counterpoint to the improved space complexity of a DFBB search instead of A* is that DFBB search cannot directly provide
                           a sorted list of suboptimal solutions. To enumerate all suboptimal solutions within <em>E</em><sub>cut</sub> = 2.0 kcal.mol<sup>−1</sup> of the GMEC, we first computed the GMEC and its associated energy <span class="inline-formula" id="inline-formula-20"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-20.gif" /></span> as above and performed a second exhaustive search for all solutions with energy below <span class="inline-formula" id="inline-formula-21"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-21.gif" /></span> (options –a and –ub in <em>toulbar2</em> to, respectively, produce all solutions and set a global upper bound).
                        </p>
                     </div>
                     
                     <div id="sec-8" class="subsection">
                        <h4>ILP model</h4>
                        
                        <p id="p-25">We also modeled the CPD problem as an integer linear program (01LP) problem using the usual translation from CFN to ILP initially
                           proposed in (<a id="xref-ref-33-1" class="xref-bibr" href="#ref-33">Koster <em>et al.</em>, 1999</a>). An ILP is defined by a linear criteria and a set of linear constraints on integer variables. For every value/rotamer i<sub>r</sub> of the variable/residue <em>i</em>, we introduced one Boolean variable <span class="inline-formula" id="inline-formula-22"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-22.gif" /></span> that indicates whether the rotamer <span class="inline-formula" id="inline-formula-23"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-23.gif" /></span> is used <span class="inline-formula" id="inline-formula-24"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-24.gif" /></span>or not <span class="inline-formula" id="inline-formula-25"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-25.gif" /></span>. To enable the expression of the energy as a linear function of variables, we introduced an extra Boolean variable <span class="inline-formula" id="inline-formula-26"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-26.gif" /></span> for every pair of rotamers <span class="inline-formula" id="inline-formula-27"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-27.gif" /></span>, capturing the fact that this pair of rotamers is used. The energy can then be expressed directly as the linear function
                           to be minimized (the constant term can be ignored as it cannot change the optimal solution):
                           <span class="disp-formula" id="disp-formula-3"><img class="graphic" alt="Formula" src="2129/embed/graphic-4.gif" /><span class="disp-formula-label">(3)</span></span>
                           
                        </p>
                        
                        <p id="p-26">Additional constraints enforce that exactly one rotamer is selected for each variable position and that a pair is used if
                           and only if the corresponding values are used. Then, finding a GMEC reduces to the following ILP:
                           <span class="disp-formula" id="disp-formula-4"><img class="graphic" alt="Formula" src="2129/embed/graphic-5.gif" /><span class="disp-formula-label">(4)</span></span>
                           
                        </p>
                        
                        <p id="p-27">such that:
                           <span class="disp-formula" id="disp-formula-5"><img class="graphic" alt="Formula" src="2129/embed/graphic-6.gif" /></span>
                           
                        </p>
                        
                        <p id="p-28">The resulting ILP contains <span class="inline-formula" id="inline-formula-28"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-28.gif" /></span> variables and <span class="inline-formula" id="inline-formula-29"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-29.gif" /></span> constraints. It is equivalent to the model proposed for CPD in (<a id="xref-ref-31-2" class="xref-bibr" href="#ref-31">Kingsford <em>et al.</em>, 2005</a>). Note that because the objective function is non-linear, it is fundamentally impossible to express it in ILP without introducing
                           a quadratic number of variables. Hence, this ILP model cannot be improved significantly in size.
                        </p>
                        
                        <p id="p-29">To identify the GMEC, we used the IBM™ ILOG ILP solver <em>cplex</em> (version 12.4.0.0) on this ILP with parameters EPAGAP, EPGAP and EPINT set to zero to avoid premature stop.
                        </p>
                     </div>
                     
                     <div id="sec-9" class="subsection">
                        <h4>Dead-end-elimination / A* combined approach (DEE/A*)</h4>
                        
                        <p id="p-30">The DEE algorithm iteratively eliminates rotamers and pairs of rotamers that cannot possibly be part of the GMEC (<a id="xref-ref-16-2" class="xref-bibr" href="#ref-16">Desmet <em>et al.</em>, 1992</a>) by using a dominance criterion. The original DEE single elimination criterion removes a rotamer <em>r</em> at position <em>i</em> if there exists another rotamer <em>u</em> at the same position such that
                           <span class="disp-formula" id="disp-formula-6"><img class="graphic" alt="Formula" src="2129/embed/graphic-7.gif" /><span class="disp-formula-label">(5)</span></span>
                           
                        </p>
                        
                        <p id="p-31">In this case, <em>r</em> can be removed because any conformation using <em>r</em> can be transformed into a lower energy conformation by substituting <em>u</em> for <em>r</em>. The pruning criterion is applied until a single solution remains (i.e. the GMEC) or all solutions outside an energy window
                           of <span class="inline-formula" id="inline-formula-30"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-30.gif" /></span> have been pruned or otherwise when no more pruning is identified during a given round. The DEE pruning step is followed by
                           an A* branch-and-bound like search, which uses the remaining rotamers (<a id="xref-ref-38-1" class="xref-bibr" href="#ref-38">Leach <em>et al.</em>, 1998</a>) to identify the GMEC or produces a sorted list of all models whose energy is within a specified energy <span class="inline-formula" id="inline-formula-31"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-31.gif" /></span> of the GMEC energy. The A* algorithm is a worst-case exponential space and exponential time algorithm whose efficiency is
                           tightly linked to the quality of the heuristic admissible evaluation function used to decide which node to explore next. Interestingly,
                           the heuristic used in the A* approach applied in this study is equivalent to the CFN heuristic used in the PFC-DAC algorithm
                           (<a id="xref-ref-49-1" class="xref-bibr" href="#ref-49">Wallace, 1996</a>). This lower bound as well as an improved variant of it (<a id="xref-ref-36-1" class="xref-bibr" href="#ref-36">Larrosa <em>et al.</em>, 1998</a>) has been obsoleted by the incremental local consistencies introduced in (<a id="xref-ref-46-2" class="xref-bibr" href="#ref-46">Schiex, 2000</a>).
                        </p>
                        
                        <p id="p-32">In this study, we used <em>Osprey 2.0</em> to perform the DEE/A* procedure to find the GMEC and suboptimal models within a <span class="inline-formula" id="inline-formula-32"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-32.gif" /></span> = 2.0 kcal.mol<sup>−1</sup> of the GMEC energy (option <span class="inline-formula" id="inline-formula-33"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-33.gif" /></span>). The procedure starts by extensive DEE (<span class="inline-formula" id="inline-formula-34"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-34.gif" /></span>, which enables simple Goldstein, Magic bullet pairs, 1- and 2-split positions, bounds and pairs pruning) and is followed
                           by the A* search. We also optionally applied a procedure including a pre-filtering step before the DEE, which eliminates rotamers
                           <span class="inline-formula" id="inline-formula-35"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-35.gif" /></span> such that <span class="inline-formula" id="inline-formula-36"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-36.gif" /></span> &gt; 30 kcal.mol<sup>−</sup><sup>1</sup> and pairs <span class="inline-formula" id="inline-formula-37"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-37.gif" /></span> such that <span class="inline-formula" id="inline-formula-38"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-38.gif" /></span> &gt; 100 kcal.mol<sup>−1</sup> (pruning and stericE parameters).
                        </p>
                        
                        <p id="p-33">All computations (<em>toulbar2</em>, <em>cplex</em> and <em>Osprey</em>) were performed on one core of an AMD Opteron™ Processor <span class="em-link"><span class="em-addr">6176{at}2.3</span></span> GHz. We used 128 GB of RAM and a 100 h time-out.
                        </p>
                     </div>
                     
                  </div>
                  
                  <div id="sec-10" class="subsection">
                     <h3>2.5 Analysis of top-score models</h3>
                     
                     <p id="p-34">For four design cases (PDB ids: 1TEN, 1UBI, 2PCY, 1CSK), the 3D structure of the best conformation of each unique sequence
                        found within a 2 kcal mol<sup>−</sup><sup>1</sup> window of the GMEC energy was built using <em>Osprey 2</em>. These 3D structure models were then subjected to 1000 steps of minimization with the <em>Sander</em> module of <em>Amber 9,</em> using the generalized born/surface area implicit solvent model. During these minimizations, heavy atom positions were restrained
                        using a harmonic potential (force constant = 1 kcal.mol<sup>−</sup><sup>1 </sup>Å<sup>−</sup><sup>2</sup>). The score of minimized structures was computed with <em>Osprey 2.0</em>. Finally, the conformational variability of the minimized models was assessed by carrying out an optimization step allowing
                        all variable amino acids (mutable and flexible) to repack. This step was performed using the CFN-based approach with a <span class="inline-formula" id="inline-formula-39"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-39.gif" /></span> of 0.2 kcal mol<sup>−</sup><sup>1</sup> and involved the pre-computing of pairwise energy terms for each minimized model using <em>Osprey 2</em>.
                     </p>
                  </div>
                  
               </div>
               <div class="section results-discussion" id="sec-11">
                  <div class="section-nav"><a href="#sec-2" title="2 METHODS" class="prev-section-link"><span>Previous Section</span></a><a href="#sec-16" title="4 CONCLUSION" class="next-section-link"><span>Next Section</span></a></div>
                  <h2>3 RESULTS AND DISCUSSION</h2>
                  
                  <div id="sec-12" class="subsection">
                     <h3>3.1 Benchmark set</h3>
                     
                     <p id="p-35">A tailored benchmark set was produced to assess the performance of combinatorial optimization algorithms on sequence-conformation
                        spaces of various sizes and complexity as well as the potential of the CPD methodology proposed herein (<a id="xref-fig-1-3" class="xref-fig" href="#F1">Fig. 1</a>) for tackling the redesign of diverse structural systems involving free proteins or proteins bound to a cofactor, a ligand
                        or a protein. The studied systems have all been extracted from previously published articles about protein engineering, <em>in silico</em> protein design or protein structural studies (see references <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>). A detailed description of our benchmark preparation protocol is given in <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Data</a>.
                     </p>
                     
                     <p id="p-36">In our benchmark set, the number of mutable residues varies from 3 to 119 (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>). They are located either in the core of (holo)proteins (except when data are available in the literature) or at the protein–protein
                        or protein–ligand interfaces. We then defined a set of flexible residues (from 1 to 93) (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>) that surrounds mutable positions and mainly occupies the core and the boundary regions of proteins. The resulting number
                        of variable residues ranges then from 23 to 120 and, given the <em>penultimate</em> rotamer library used, from 3 to 194 amino acid rotamers were considered at each variable position. Our resulting benchmark
                        set covers thus a wide range of combinatorial spaces (from 4.10<sup>26</sup> to 2.10<sup>249</sup>) and allows us to evaluate different combinatorial optimization problems of varying complexity.
                     </p>
                  </div>
                  
                  <div id="sec-13" class="subsection">
                     <h3>3.2 GMEC-based design</h3>
                     
                     <p id="p-37">First, we evaluated the performance of CFN and ILP methods for solving the GMEC identification problem exactly and compared
                        them against the exact CPD-specific method DEE/A*. We compared the CFN solver <em>toulbar2</em>, the ILP solver <em>cplex</em> and the DEE/A* implemented in the <em>Osprey</em> software on the benchmark set of 35 design cases (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>) and present the results in <a id="xref-table-wrap-1-1" class="xref-table" href="#T1">Table 1</a>.
                        
                     </p>
                     <div class="table pos-float" id="T1">
                        <div class="table-inline">
                           <div class="callout"><span>View this table:</span><ul class="callout-links">
                                 <li><a href="2129/T1.expansion.html">In this window</a></li>
                                 <li><a class="in-nw" href="2129/T1.expansion.html">In a new window</a></li>
                              </ul>
                           </div>
                        </div>
                        <div class="table-caption"><span class="table-label">Table 1.</span> 
                           <p id="p-38" class="first-child">CPU-time for solving the GMEC using DEE/A* (<em>osprey</em>), ILP (<em>cplex</em>) and CFN (<em>toulbar2</em>)
                           </p>
                           <div class="sb-div caption-clear"></div>
                        </div>
                     </div>
                     
                     <p id="p-40">Out of the 35 design cases, the CFN solver <em>toulbar2</em> and the ILP solver cplex solved, respectively, 30 and 27 cases within the 100 h time-out, whereas the DEE/A* CPD-dedicated
                        approach identified only 18 GMEC (<a id="xref-table-wrap-1-2" class="xref-table" href="#T1">Table 1</a>).
                     </p>
                     
                     <p id="p-41">The DEE/A* method managed to find the 18 GMEC with CPU times ranging from a few minutes to &gt;1 h. Only four cases (1MJC, 1CSK,
                        1SHF and 1NXB) corresponding to the exploration of small combinatorial spaces on small proteins took &lt;1 min to be solved.
                        The DEE step converged to a single solution in only three instances (1MJC, 1NXB, 1CSE) (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S3</a>). The A* search successfully identified 15 further GMEC using the subset of rotamers remaining after the DEE step. It failed
                        for 12 instances owing to time limit (10 cases) or memory limit (2 cases). In these cases, the size of the combinatorial search
                        space of non-pruned rotamers after the DEE step ranged ∼10<sup>19</sup> to 10<sup>36</sup>. This was not a sufficient reduction to enable A* to extract the GMEC from the subset of remaining rotamers with the available
                        computational resources. <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S3</a> gives fractions of times spent during DEE and A* for each instance. Using <em>toulbar2</em> instead of A* after the DEE step allowed to quickly identify the GMEC for the 12 DEE/A* unsolvable instances. A* is therefore
                        the limiting step for these 12 instances. Finally, the DEE step failed to complete within the 100 h time-out for five tests
                        covering initial combinatorial spaces from ∼10<sup>61</sup> to 10<sup>249</sup> and the highest number of variable residues at the surface of the proteins. One can expect the surface residues to be more
                        flexible than the buried residues, and such flexibility may then lead to a high density of conformations with similar energy
                        and a corresponding increase in the complexity of the combinatorial space to be explored. These results clearly underline
                        the limits of the DEE-based approach to handle large and complex problems.
                     </p>
                     
                     <p id="p-42">To improve the efficiency of the DEE-based approach, a pre-processing is usually applied to eliminate rotamers and pairs of
                        rotamers of high energy, which are not expected to appear in the GMEC. We then performed such pre-filtering using a 30 kcal
                        mol<sup>−</sup><sup>1</sup> threshold for rotamers and a 100 kcal mol<sup>−</sup><sup>1</sup> threshold for pairs of rotamers. Using these parameters in one instance (1CSE), the optimal solution after preprocessing
                        did not match the GMEC obtained in the absence of this pre-processing step. However, when we increased the threshold for rotamers
                        30–50 kcal mol<sup>−</sup><sup>1</sup>, the optimal solution remained the same with and without preprocessing. Besides losing the guarantee of optimality, the preprocessing
                        step did not improve the performance of DEE/A* in terms of the number of instances that were solved (18 GMEC identified out
                        of the 35 cases) (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S3</a>). Furthermore, seven instances required more CPU time for identifying the GMEC when preprocessing was used.
                     </p>
                     
                     <p id="p-43">The ILP solver <em>cplex</em> solved nine additional instances from our benchmark set (2DRI, 1UBI, 1CTF, 1CM1, 1BRS, 1LZ1, 2RN2, 3CHY, 1L63) compared with
                        DEE/A* (<a id="xref-table-wrap-1-3" class="xref-table" href="#T1">Table 1</a>). In all these cases, DEE/A* timed out during the A* search. Among these nine design cases, only two (3CHY and 1L63) cover
                        combinatorial spaces of greater size (∼10<sup>92</sup> and 10<sup>94</sup>) than the largest combinatorial problem (1HNG ∼10<sup>88</sup>) solved by DEE/A* (<a id="xref-table-wrap-1-4" class="xref-table" href="#T1">Table 1</a>). The other 18 instances solved by <em>cplex</em> are the ones that were successfully solved by DEE/A*. Although the ILP solver was faster in 13 of these 18 instances, the
                        time is overall similar for both methods. Nevertheless, the total number of solved instances shows that the ILP solvers can
                        be more efficient for several design cases, as previously reported (<a id="xref-ref-2-1" class="xref-bibr" href="#ref-2">Allouche <em>et al.</em>, 2012</a>). More concise quadratic programming (cplex QP solver) and partial weighted MaxSAT (akmaxsat, MaxHS and Bin-Core-Dis solvers)
                        models were also tried, but they all failed on all instances.
                     </p>
                     
                     <p id="p-44">The CFN solver <em>toulbar2</em> solved, respectively, 12 and 3 more cases compared with DEE/A* and <em>cplex</em> (<a id="xref-table-wrap-1-5" class="xref-table" href="#T1">Table 1</a>). Therefore, CFN only failed on five instances (1PGB, 1ENH, 2CI2, 3HHR, 1STN) out of the 35 handled. These instances correspond
                        to vast combinatorial spaces (from about 10<sup>61</sup> to 10<sup>249</sup>), which mostly include variable residues scattered over the three layers of the proteins (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>). There are no cases solved by DEE/A* or ILP that CFN could not solve. Moreover, CFN outperformed the two other approaches
                        by an impressive margin in terms of speed. Among the 30 instances solved by CFN and including large combinatorial spaces,
                        11 cases were solved in &lt;1 s, 23 in &lt;10 s and only five instances required a few minutes. Given its running time performance
                        and its success rate for handling large protein design problems, the CFN approach appears as an appealing alternative to current
                        exact CPD-dedicated methods, especially for solving highly complex GMEC-based design problems.
                     </p>
                     
                     <p id="p-45">There is no single explanation for the performance advantage of the CFN solver <em>toulbar2</em> over the ILP solver <em>cplex</em> and the DEE/A* implemented in the <em>Osprey</em>. Indeed, solvers are complex systems involving various mechanisms. The effect of their interactions during solving is hard
                        to predict. Moreover, the IBM™ ILOG ILP solver cplex is a totally closed-source industrial black box solver.
                     </p>
                     
                     <p id="p-46">Compared with the CFN solver <em>toulbar2</em>, <em>Osprey</em> uses a lower bound considered as obsolete in CFN (<a id="xref-ref-49-2" class="xref-bibr" href="#ref-49">Wallace, 1996</a>) instead of the recent incremental stronger lower bounds offered by soft local consistencies such as EDAC (<a id="xref-ref-37-1" class="xref-bibr" href="#ref-37">Larrosa <em>et al.</em>, 2005</a>). This, together with the associated informed value ordering provided by these local consistencies, explains why the CFN
                        approach outperforms the DEE/A* method. Considering ILP, it is known that the LP relaxation lower bound used in ILP is (by
                        duality) the same as the Optimal Soft AC (<a id="xref-ref-11-1" class="xref-bibr" href="#ref-11">Cooper <em>et al.</em>, 2010</a>) lower bound when no upper bounding occurs. Because OSAC dominates all other local consistencies at the arc level, this provides
                        an explanation for the efficiency of <em>cplex</em> compared with <em>Osprey</em>. Finally, compared with ILP, the formulation of the deeply non-linear problem is more direct in CFN. This likely contributes,
                        together with the upper bounding, variable and value ordering heuristics of <em>toulbar2</em>, to the efficiency of the CFN approach compared with ILP.
                     </p>
                  </div>
                  
                  <div id="sec-14" class="subsection">
                     <h3>3.3 Suboptimal ensemble generation</h3>
                     
                     <p id="p-47">We also performed computational design tests to assess the ability of the CFN method to generate an ensemble of provably near-optimal
                        sequence-conformation models in addition to the GMEC. The performance of the CFN approach was compared against DEE/A*. For
                        this purpose, the 35 design cases of the benchmark set (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>) were again investigated using the CFN <em>toulbar2</em> solver and the DEE/A* implemented in <em>Osprey</em> software to access the set of sequence-conformation models comprised within an energy window of 2 kcal mol<sup>−</sup><sup>1</sup> of the GMEC energy.
                     </p>
                     
                     <p id="p-48">Out of the 35 design cases, the CFN solver <em>toulbar2</em> managed to produce the suboptimal ensembles of solutions for 30 design cases, whereas the DEE/A* approach only successfully
                        handled one instance (<a id="xref-table-wrap-2-1" class="xref-table" href="#T2">Table 2</a>).
                        
                     </p>
                     <div class="table pos-float" id="T2">
                        <div class="table-inline">
                           <div class="callout"><span>View this table:</span><ul class="callout-links">
                                 <li><a href="2129/T2.expansion.html">In this window</a></li>
                                 <li><a class="in-nw" href="2129/T2.expansion.html">In a new window</a></li>
                              </ul>
                           </div>
                        </div>
                        <div class="table-caption"><span class="table-label">Table 2.</span> 
                           <p id="p-49" class="first-child">CPU-time for generating the ensemble of suboptimal models (E<em><sub>cut</sub></em> = 2 kcal.mol<sup>−1</sup>) using DEE/A* (<em>osprey</em>) and CFN (<em>toulbar2</em>)
                           </p>
                           <div class="sb-div caption-clear"></div>
                        </div>
                     </div>
                     
                     <p id="p-51">The DEE/A* approach failed for 34 instances owing to time (30 cases) or memory (4 cases) limits. It only identified the set
                        of near-optimal models for one instance (1SHF) among the 18 successfully handled for the GMEC problem (<a id="xref-table-wrap-1-6" class="xref-table" href="#T1">Table 1</a>). Although this solved instance corresponds to one of the smallest investigated combinatorial spaces (∼10<sup>34</sup>) (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>), ∼37 h of computation were needed to find the ensemble of low-energy models compared with &lt;1 s for the CFN approach. This
                        running time is even &gt;∼7 h required by the CFN method in the worst case (1L63) including an important combinatorial space
                        (∼10<sup>94</sup>) and a large number of suboptimal solutions (8 × 10<sup>8</sup>).
                     </p>
                     
                     <p id="p-52">The CFN method only failed to identify the near-optimal ensemble on the five instances (1PGB, 1ENH, 2CI2, 3HHR, 1STN) for
                        which the GMEC problem was also unsolved (<a id="xref-table-wrap-1-7" class="xref-table" href="#T1">Table 1</a>). In addition to the high success rate achieved by CFN, the method was also efficient: 10 cases were solved by CFN in &lt;1
                        min, 15 required several minutes and only five instances needed several hours (<a id="xref-table-wrap-2-2" class="xref-table" href="#T2">Table 2</a>).
                     </p>
                     
                     <p id="p-53">While the task of finding a set of low-energy sequence-conformation models proved to be an insurmountable computational hurdle
                        for DEE/A* as implemented in <em>Osprey</em>, the CFN solver <em>toulbar2</em> successfully solved most of the design cases tested. Moreover, the CFN approach gave access to sets of provably suboptimal
                        solutions with outstanding running time performances.
                     </p>
                     
                     <p id="p-54">The CFN approach efficiently uses the knowledge of the GMEC solution in the enumeration procedure of the near-optimal models.
                        The GMEC defines an upper bound corresponding to the energy of the GMEC + 2 kcal mol<sup>−</sup><sup>1</sup>. In CFN, this upper bound is systematically compared with the lower bound provided by local consistency enforcing. The DEE
                        implemented in <em>Osprey</em> uses the same upper bound (parameter pruningE) but exploits a weaker lower bound. This likely explains the performance gap
                        compared with the <em>toulbar2</em> CFN solver.
                     </p>
                  </div>
                  
                  <div id="sec-15" class="subsection">
                     <h3>3.4 Suboptimal ensemble analysis</h3>
                     
                     <p id="p-55">First, we analyzed the sequence and conformational variability of the near-optimal models obtained for four design cases (1CSK,
                        1TEN, 1UBI, 2PCY) of proteins adopting distinct structural folds (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S2</a>). These instances include from 9 to 16 mutable residues and from 21 to 30 flexible residues (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Table S1</a>).
                     </p>
                     
                     <p id="p-56">Within a window of 2 kcal mol<sup>−</sup><sup>1</sup> of the GMEC, the CFN-based approach produced &gt;10<sup>5</sup> sequence-conformation models for each of the four design cases (<a id="xref-table-wrap-2-3" class="xref-table" href="#T2">Table 2</a>). The score of these models is lower by as much as ∼20 kcal mol<sup>−</sup><sup>1</sup> than that of the wild-type model (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Figs S2c–S5c</a>). In these ensembles of models, 144, 194, 199 and 294 unique sequences were found, respectively, for 2PCY, 1UBI, 1CSK and
                        1TEN design cases.
                     </p>
                     
                     <p id="p-57">Only few unique sequences were then generated compared with the high number of enumerated models within a small energy window
                        of 2 kcal mol<sup>−1</sup> of the GMEC energy. However, when the experimental construction of the protein library is considered, it is important to
                        have access to a larger ensemble of distinct sequences. For this purpose, the outstanding performances of the CFN solver (<a id="xref-table-wrap-2-4" class="xref-table" href="#T2">Table 2</a>) could be harnessed to provably predict suboptimal models distributed on a wider energy window of the GMEC and thus attempt
                        to generate more diverse sequence ensembles.
                     </p>
                     
                     <p id="p-58">For the four design cases, the wild-type amino acid was the most often either substituted by an amino acid of slightly larger
                        size or conserved (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Figs S2a–S5a</a>). Nevertheless, the entire wild-type sequence of the protein was never found within the suboptimal ensembles. It is noteworthy
                        that the mutable residues of glycine type were not found substituted by another amino acid type (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Figs S2a</a>, <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">S3a</a> and <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">S4a</a>). The number of conformations adopted by each sequence decreases gradually as the energy value of the sequence becomes more
                        unfavorable (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Figs S2d–S5d</a>). The superposition of the best conformation of each unique sequence showed that each flexible residue adopts almost the
                        same rotamer in all best conformations (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Figs S2b–S5b</a>). Moreover, the orientation of mutable residue side-chains is similar in all the best models regardless the assigned amino-acid
                        type.
                     </p>
                     
                     <p id="p-59">For the protein design problems studied here, we expected that mutations would favor the introduction of bulkier amino acids
                        to fill up the free space available in the core of proteins. However, changes in amino acid sizes were subtle. A visual inspection
                        of the 3D structures of mutants suggests that some slight adjustments of side chains and/or backbone of surrounding residues
                        could enable accommodation of larger side chains, which were here assigned with high interaction energies. This lack of conformational
                        relaxation seems also to be at the origin of the observed conservation of glycine amino acid types. Therefore, the sequence
                        selection may be biased and restricted by the lack of flexibility of surrounding residues. These results highlight the key
                        role of the local molecular flexibility to extend the accessible sequence space, as shown by previous work (<a id="xref-ref-7-1" class="xref-bibr" href="#ref-7">Bordner and Abagyan, 2004</a>). We then subjected each unique sequence of the four suboptimal ensembles to energy minimization to assess the effect of
                        the relaxation of side chains and backbone freedom degrees on the energy ranking of the sequence ensembles. Overall, the minimization
                        decreased the energy values of these models from ∼20 to 60 kcal mol<sup>−</sup><sup>1</sup> depending on the design case (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Figs S2c–S5c</a>). Nonetheless, the superposition of the structures before and after minimization only showed slight rearrangements of protein
                        side chains and backbone. This clearly indicates that slight geometrical adjustments can significantly lower model energies.
                        The conformational variability of these low-energy sequences were further investigated by carrying out an optimization step
                        (with a <span class="inline-formula" id="inline-formula-40"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-40.gif" /></span> of 0.2 kcal mol<sup>−</sup><sup>1</sup>), which enables all variable amino acids (mutable and flexible) to be repacked. Despite an <span class="inline-formula" id="inline-formula-41"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-41.gif" /></span> value, which is 10 times smaller, the number of conformations adopted by each unique sequence was found extremely high whatever
                        the energy ranking of the sequence (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Figs S2d–S5d</a>). Therefore, the significant differences observed among mutants before minimization step, is probably the result of the discretization
                        of conformational freedom degrees. The minimization step thus allows us to extend the accessible conformational space. Current
                        trends in CPD refine the exact DEE/A* approach along various directions, allowing, respectively, for continuous rotamers (<a id="xref-ref-20-1" class="xref-bibr" href="#ref-20">Georgiev <em>et al.</em>, 2008a</a>,<a id="xref-ref-21-1" class="xref-bibr" href="#ref-21">b</a>; <a id="xref-ref-18-2" class="xref-bibr" href="#ref-18">Gainza <em>et al.</em>, 2012</a>), for continuous (<a id="xref-ref-19-1" class="xref-bibr" href="#ref-19">Georgiev and Donald, 2007</a>) or discrete (<a id="xref-ref-20-2" class="xref-bibr" href="#ref-20">Georgiev <em>et al.</em>, 2008a</a>,<a id="xref-ref-21-2" class="xref-bibr" href="#ref-21">b</a>) backbone conformation adjustments or both (<a id="xref-ref-24-2" class="xref-bibr" href="#ref-24">Hallen <em>et al.</em>, 2013</a>). The CFN approach could further be extended to handle such flexibilities descriptions.
                     </p>
                     
                     <p id="p-60">In addition to lowering the energy and increasing the conformational variability of models, the geometry relaxation step reranks
                        the sequence ensemble. The GMEC obtained after minimization (refined GMEC) does not match with the original GMEC. The energy
                        values of the minimized suboptimal models are spread within an energy window up to ∼6 kcal mol<sup>−</sup><sup>1</sup> of the refined GMEC (<a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/lookup/suppl/doi:10.1093/bioinformatics/btt374/-/DC1">Supplementary Figs S2c–S5c</a>). Therefore, with a <span class="inline-formula" id="inline-formula-42"><img class="inline-graphic" alt="Graphic" src="2129/embed/inline-graphic-42.gif" /></span> of 2 kcal mol<sup>−</sup><sup>1</sup>, from 92 to 185 unique sequences depending on the design case would be removed of these minimized ensembles. Even on a small
                        energy window, these results demonstrate the advantage of the post-minimization to re-rank and post-screen the most promising
                        candidate sequences to evaluate experimentally.
                     </p>
                  </div>
                  
               </div>
               <div class="section conclusions" id="sec-16">
                  <div class="section-nav"><a href="#sec-11" title="3 RESULTS AND DISCUSSION" class="prev-section-link"><span>Previous Section</span></a><a href="#ack-1" title="ACKNOWLEDGEMENTS" class="next-section-link"><span>Next Section</span></a></div>
                  <h2>4 CONCLUSION</h2>
                  
                  <p id="p-61">In this article, we have formulated a novel open-source based computational framework to provably identify the GMEC as well
                     as a set of low-energy protein sequences within the context of atomic protein design. This CFN-based approach provides remarkable
                     speedups, allowing us to explore vast sequence-conformational spaces much more efficiently than the DEE/A* algorithm or state-of-the-art
                     ILP algorithms. Despite the significant change in terms of problem complexity, it is surprising to see that this efficiency
                     extends to the generation of gap-free sets of suboptimal solutions. This article and the companion open-source computational
                     tools we offer will therefore facilitate the optimization of new CPD systems, without requiring expensive computational resources.
                  </p>
                  
                  <p id="p-62">Ultimately, we hope that CFN technology will allow complex CPD problems, mixing optimization of flexible systems and discrete
                     integration (capturing entropic effects and affinity) to be directly tackled.
                  </p>
               </div>
               <div class="section ack" id="ack-1">
                  <div class="section-nav"><a href="#sec-16" title="4 CONCLUSION" class="prev-section-link"><span>Previous Section</span></a><a href="#fn-group-1" title="Footnotes" class="next-section-link"><span>Next Section</span></a></div>
                  <h2>ACKNOWLEDGEMENTS</h2>
                  
                  <p id="p-63">We thank the Computing Center of Region Midi-Pyrénées (CALMIP, Toulouse, France) and the GenoToul Bioinformatics Platform
                     of INRA-Toulouse for providing computing resources and support.
                  </p>
                  
                  <p id="p-64"><em>Funding</em>: <span class="funding-source">Agence Nationale de la Recherche</span> (<span class="award-id" id="award-id-1">ANR 10-BLA-0214</span>, <span class="award-id" id="award-id-2">ANR-12-MONU-0015-03</span>); <span class="funding-source">INRA and the Region Midi-Pyrénées</span> (to S.T.).
                  </p>
                  
                  <p id="p-65"><em>Conflict of Interest</em>: none declared.
                  </p>
               </div>
               <div class="section fn-group" id="fn-group-1">
                  <div class="section-nav"><a href="#ack-1" title="ACKNOWLEDGEMENTS" class="prev-section-link"><span>Previous Section</span></a><a href="#ref-list-1" title="REFERENCES" class="next-section-link"><span>Next Section</span></a></div>
                  <h2>Footnotes</h2>
                  <ul>
                     <li class="fn" id="fn-1">
                        <p id="p-1"><a class="rev-xref" href="#xref-fn-1-1">↵</a><sup>†</sup>The authors wish it to be known that, in their opinion, the first two authors should be regarded as joint First Authors.
                        </p>
                     </li>
                     <li class="fn" id="fn-2">
                        <p id="p-2">Associate Editor: Anna Tramontano</p>
                     </li>
                  </ul>
               </div>
               <ul class="copyright-statement">
                  <li class="fn" id="copyright-statement-1">© The Author 2013. Published by Oxford University Press. All rights reserved. For Permissions, please e-mail: journals.permissions@oup.com</li>
               </ul>
               <div class="section ref-list" id="ref-list-1">
                  <div class="section-nav"><a href="#fn-group-1" title="Footnotes" class="prev-section-link"><span>Previous Section</span></a><div class="nav-placeholder"> </div>
                  </div>
                  <h2>REFERENCES</h2>
                  <ol class="cit-list ref-use-labels">
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-1-1" title="View reference  in text"
                           id="ref-1">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.1">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Allen</span>  <span class="cit-name-given-names">BD</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Mayo</span>  <span class="cit-name-given-names">SL</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Dramatic performance enhancements for the FASTER optimization algorithm</span>. <abbr class="cit-jnl-abbrev">J. Comput. Chem.</abbr> <span class="cit-pub-date">2006</span>;<span class="cit-vol">27</span>:<span class="cit-fpage">1071</span>-<span class="cit-lpage">1075</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/jcc.20420&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=16685715&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000238158900003&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-2-1" title="View reference  in text"
                           id="ref-2">↵</a><div class="cit ref-cit ref-book" id="cit-29.17.2129.2">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Allouche</span>  <span class="cit-name-given-names">D</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-source">Proceedings of Principles and Practice of Constraint Programming–CP 2012, Québec City, QC, Canada</span>. <span class="cit-pub-date">2012</span>. <span class="cit-article-title">Computational protein design as a cost function network optimization problem</span>; p. <span class="cit-fpage">840</span>-<span class="cit-lpage">849</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Computational%20protein%20design%20as%20a%20cost%20function%20network%20optimization%20problem&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Allouche&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-3-1" title="View reference  in text"
                           id="ref-3">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.3">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Althaus</span>  <span class="cit-name-given-names">E</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">A combinatorial approach to protein docking with flexible side chains</span>. <abbr class="cit-jnl-abbrev">J. Comput. Biol.</abbr> <span class="cit-pub-date">2002</span>;<span class="cit-vol">9</span>:<span class="cit-fpage">597</span>-<span class="cit-lpage">612</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1089/106652702760277336&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=12323095&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000177959800001&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-4-1" title="View reference  in text"
                           id="ref-4">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.4">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Archontis</span>  <span class="cit-name-given-names">G</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Simonson</span>  <span class="cit-name-given-names">T</span></span></li>
                              </ol><cite>. <span class="cit-article-title">A residue-pairwise generalized born scheme suitable for protein design calculations</span>. <abbr class="cit-jnl-abbrev">J. Phys. Chem. B</abbr> <span class="cit-pub-date">2005</span>;<span class="cit-vol">109</span>:<span class="cit-fpage">22667</span>-<span class="cit-lpage">22673</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=16853951&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-5-1" title="View reference  in text"
                           id="ref-5">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.5">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Arnold</span>  <span class="cit-name-given-names">FH</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Combinatorial and computational challenges for biocatalyst design</span>. <abbr class="cit-jnl-abbrev">Nature</abbr> <span class="cit-pub-date">2001</span>;<span class="cit-vol">409</span>:<span class="cit-fpage">253</span>-<span class="cit-lpage">257</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1038/35051731&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=11196654&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-6-1" title="View reference  in text"
                           id="ref-6">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.6">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Bernstein</span>  <span class="cit-name-given-names">FC</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">The Protein Data Bank. A computer-based archival file for macromolecular structures</span>. <abbr class="cit-jnl-abbrev">Eur. J. Biochem.</abbr> <span class="cit-pub-date">1977</span>;<span class="cit-vol">80</span>:<span class="cit-fpage">319</span>-<span class="cit-lpage">324</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=923582&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=A1977EB89500001&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-7-1" title="View reference  in text"
                           id="ref-7">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.7">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Bordner</span>  <span class="cit-name-given-names">AJ</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Abagyan</span>  <span class="cit-name-given-names">RA</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Large-scale prediction of protein geometry and stability changes for arbitrary single point mutations</span>. <abbr class="cit-jnl-abbrev">Proteins</abbr> <span class="cit-pub-date">2004</span>;<span class="cit-vol">57</span>:<span class="cit-fpage">400</span>-<span class="cit-lpage">413</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/prot.20185&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=15340927&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000223926200017&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-8-1" title="View reference  in text"
                           id="ref-8">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.8">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Cabon</span>  <span class="cit-name-given-names">B</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Radio link frequency assignment</span>. <abbr class="cit-jnl-abbrev">Constraints</abbr> <span class="cit-pub-date">1999</span>;<span class="cit-vol">4</span>:<span class="cit-fpage">79</span>-<span class="cit-lpage">89</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1023/A:1009812409930&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-9-1" title="View reference  in text"
                           id="ref-9">↵</a><div class="cit ref-cit ref-book" id="cit-29.17.2129.9">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Case</span>  <span class="cit-name-given-names">DA</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-source">AMBER 9</span>. <span class="cit-publ-loc">USA</span>: <span class="cit-publ-name">University of California, San Francisco, CA</span>; <span class="cit-pub-date">2006</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=AMBER%209&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Case&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-10-1" title="View reference  in text"
                           id="ref-10">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.10">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Chen</span>  <span class="cit-name-given-names">CY</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Computational structure-based redesign of enzyme activity</span>. <abbr class="cit-jnl-abbrev">Proc. Natl Acad. Sci. USA</abbr> <span class="cit-pub-date">2009</span>;<span class="cit-vol">106</span>:<span class="cit-fpage">3764</span>-<span class="cit-lpage">3769</span>.</cite></div>
                           <div class="cit-extra"><a href="/cgi/ijlink?linkType=ABST&amp;journalCode=pnas&amp;resid=106/10/3764"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-ijlinks"><span class="cit-reflinks-abstract">Abstract</span><span class="cit-sep cit-reflinks-variant-name-sep">/</span><span class="cit-reflinks-full-text"><span class="free-full-text">FREE </span>Full Text</span></a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-11-1" title="View reference  in text"
                           id="ref-11">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.11">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Cooper</span>  <span class="cit-name-given-names">MC</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Soft arc consistency revisited</span>. <abbr class="cit-jnl-abbrev">Artif. Intell</abbr> <span class="cit-pub-date">2010</span>;<span class="cit-vol">174</span>:<span class="cit-fpage">449</span>-<span class="cit-lpage">478</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1016/j.artint.2010.02.001&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=000277328200001&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-12-1" title="View reference  in text"
                           id="ref-12">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.12">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Dahiyat</span>  <span class="cit-name-given-names">BI</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Mayo</span>  <span class="cit-name-given-names">SL</span></span></li>
                              </ol><cite>. <span class="cit-article-title">De novo protein design: fully automated sequence selection</span>. <abbr class="cit-jnl-abbrev">Science</abbr> <span class="cit-pub-date">1997</span>;<span class="cit-vol">278</span>:<span class="cit-fpage">82</span>-<span class="cit-lpage">87</span>.</cite></div>
                           <div class="cit-extra"><a href="/cgi/ijlink?linkType=ABST&amp;journalCode=sci&amp;resid=278/5335/82"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-ijlinks"><span class="cit-reflinks-abstract">Abstract</span><span class="cit-sep cit-reflinks-variant-name-sep">/</span><span class="cit-reflinks-full-text"><span class="free-full-text">FREE </span>Full Text</span></a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-13-1" title="View reference  in text"
                           id="ref-13">↵</a><div class="cit ref-cit ref-book" id="cit-29.17.2129.13">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">De Givry</span>  <span class="cit-name-given-names">S</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-source">Proceedings of 19th International Joint Conference on Artificial Intelligence</span>. <span class="cit-publ-loc">USA</span>: <span class="cit-publ-name">San Francisco, CA</span>; <span class="cit-pub-date">2005</span>. <span class="cit-article-title">Existential arc consistency: getting closer to full arc consistency in weighted CSPs</span>; p. <span class="cit-fpage">84</span>-<span class="cit-lpage">89</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Existential%20arc%20consistency%3A%20getting%20closer%20to%20full%20arc%20consistency%20in%20weighted%20CSPs&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=De%20Givry&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-14-1" title="View reference  in text"
                           id="ref-14">↵</a><div class="cit ref-cit ref-book" id="cit-29.17.2129.14">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">De Givry</span>  <span class="cit-name-given-names">S</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-source">Proceedings of 8th World Congress on Genetics Applied to Livestock Production</span>. <span class="cit-vol">Vol. 2</span>. <span class="cit-publ-loc">Brazil</span>: <span class="cit-publ-name">Belo Horizonte</span>; <span class="cit-pub-date">2006</span>. <span class="cit-article-title">Mendelsoft: Mendelian error detection in complex pedigree using weighted constraint satisfaction techniques</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Mendelsoft%3A%20Mendelian%20error%20detection%20in%20complex%20pedigree%20using%20weighted%20constraint%20satisfaction%20techniques&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=De%20Givry&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-15-1" title="View reference  in text"
                           id="ref-15">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.15">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Desjarlais</span>  <span class="cit-name-given-names">JR</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Handel</span>  <span class="cit-name-given-names">TM</span></span></li>
                              </ol><cite>. <span class="cit-article-title">De novo design of the hydrophobic cores of proteins</span>. <abbr class="cit-jnl-abbrev">Protein Sci.</abbr> <span class="cit-pub-date">1995</span>;<span class="cit-vol">4</span>:<span class="cit-fpage">2006</span>-<span class="cit-lpage">2018</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=8535237&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=A1995RZ76600006&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-16-1" title="View reference  in text"
                           id="ref-16">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.16">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Desmet</span>  <span class="cit-name-given-names">J</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">The dead-end elimination theorem and its use in protein sidechain positioning</span>. <abbr class="cit-jnl-abbrev">Nature</abbr> <span class="cit-pub-date">1992</span>;<span class="cit-vol">356</span>:<span class="cit-fpage">539</span>-<span class="cit-lpage">542</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1038/356539a0&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=21488406&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=A1992HN23100065&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-17-1" title="View reference  in text"
                           id="ref-17">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.17">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Desmet</span>  <span class="cit-name-given-names">J</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Fast and accurate side-chain topology and energy refinement (FASTER) as a new method for protein structure optimization</span>. <abbr class="cit-jnl-abbrev">Proteins</abbr> <span class="cit-pub-date">2002</span>;<span class="cit-vol">48</span>:<span class="cit-fpage">31</span>-<span class="cit-lpage">43</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/prot.10131&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=12012335&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000176002100003&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-18-1" title="View reference  in text"
                           id="ref-18">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.18">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Gainza</span>  <span class="cit-name-given-names">P</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Protein design using continuous rotamers</span>. <abbr class="cit-jnl-abbrev">PLoS Comput. Biol.</abbr> <span class="cit-pub-date">2012</span>;<span class="cit-vol">8</span>:<span class="cit-fpage">e1002335</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1371/journal.pcbi.1002335&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=22279426&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-19-1" title="View reference  in text"
                           id="ref-19">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.19">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Georgiev</span>  <span class="cit-name-given-names">I</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Donald</span>  <span class="cit-name-given-names">BR</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Dead-end elimination with backbone flexibility</span>. <abbr class="cit-jnl-abbrev">Bioinformatics</abbr> <span class="cit-pub-date">2007</span>;<span class="cit-vol">23</span>:<span class="cit-fpage">i185</span>-<span class="cit-lpage">i194</span>.</cite></div>
                           <div class="cit-extra"><a href="/cgi/ijlink?linkType=ABST&amp;journalCode=bioinfo&amp;resid=23/13/i185"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-ijlinks"><span class="cit-reflinks-abstract">Abstract</span><span class="cit-sep cit-reflinks-variant-name-sep">/</span><span class="cit-reflinks-full-text"><span class="free-full-text">FREE </span>Full Text</span></a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-20-1" title="View reference  in text"
                           id="ref-20">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.20">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Georgiev</span>  <span class="cit-name-given-names">I</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">The minimized dead-end elimination criterion and its application to protein redesign in a hybrid scoring and search algorithm
                                    for computing partition functions over molecular ensembles</span>. <abbr class="cit-jnl-abbrev">J. Comput. Chem.</abbr> <span class="cit-pub-date">2008a</span>;<span class="cit-vol">29</span>:<span class="cit-fpage">1527</span>-<span class="cit-lpage">1542</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/jcc.20909&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=18293294&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000257038200001&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-21-1" title="View reference  in text"
                           id="ref-21">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.21">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Georgiev</span>  <span class="cit-name-given-names">I</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Algorithm for backrub motions in protein design</span>. <abbr class="cit-jnl-abbrev">Bioinfmatics</abbr> <span class="cit-pub-date">2008b</span>;<span class="cit-vol">24</span>:<span class="cit-fpage">i196</span>-<span class="cit-lpage">i204</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1093/bioinformatics/btn169&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-22-1" title="View reference  in text"
                           id="ref-22">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.22">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Gordon</span>  <span class="cit-name-given-names">DB</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Mayo</span>  <span class="cit-name-given-names">SL</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Branch-and-terminate: a combinatorial optimization algorithm for protein design</span>. <abbr class="cit-jnl-abbrev">Structure</abbr> <span class="cit-pub-date">1999</span>;<span class="cit-vol">7</span>:<span class="cit-fpage">1089</span>-<span class="cit-lpage">1098</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10508778&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-23-1" title="View reference  in text"
                           id="ref-23">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.23">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Grunwald</span>  <span class="cit-name-given-names">I</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Mimicking biopolymers on a molecular scale: nano(bio)technology based on engineered proteins</span>. <abbr class="cit-jnl-abbrev">Philos. Trans. A Math. Phys. Eng. Sci.</abbr> <span class="cit-pub-date">2009</span>;<span class="cit-vol">367</span>:<span class="cit-fpage">1727</span>-<span class="cit-lpage">1747</span>.</cite></div>
                           <div class="cit-extra"><a href="/cgi/ijlink?linkType=ABST&amp;journalCode=roypta&amp;resid=367/1894/1727"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-ijlinks"><span class="cit-reflinks-abstract">Abstract</span><span class="cit-sep cit-reflinks-variant-name-sep">/</span><span class="cit-reflinks-full-text"><span class="free-full-text">FREE </span>Full Text</span></a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-24-1" title="View reference  in text"
                           id="ref-24">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.24">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Hallen</span>  <span class="cit-name-given-names">MA</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Dead-end elimination with perturbations (DEEPer): a provable protein design algorithm with continuous sidechain and backbone
                                    flexibility</span>. <abbr class="cit-jnl-abbrev">Proteins</abbr> <span class="cit-pub-date">2013</span>;<span class="cit-vol">81</span>:<span class="cit-fpage">18</span>-<span class="cit-lpage">39</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/prot.24150&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=22821798&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000312551700002&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-25-1" title="View reference  in text"
                           id="ref-25">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.25">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Hawkins</span>  <span class="cit-name-given-names">GD</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Parametrized models of aqueous free energies of solvation based on pairwise descreening of solute atomic charges from a dielectric
                                    medium</span>. <abbr class="cit-jnl-abbrev">J. Phys. Chem.</abbr> <span class="cit-pub-date">1996</span>;<span class="cit-vol">100</span>:<span class="cit-fpage">19824</span>-<span class="cit-lpage">19839</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1021/jp961710n&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=A1996VZ91200014&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-26-1" title="View reference  in text"
                           id="ref-26">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.26">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Hellinga</span>  <span class="cit-name-given-names">HW</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Richards</span>  <span class="cit-name-given-names">FM</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Construction of new ligand binding sites in proteins of known structure: I. Computer-aided modeling of sites with pre-defined
                                    geometry</span>. <abbr class="cit-jnl-abbrev">J. Mol. Biol.</abbr> <span class="cit-pub-date">1991</span>;<span class="cit-vol">222</span>:<span class="cit-fpage">763</span>-<span class="cit-lpage">785</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1016/0022-2836(91)90510-D&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=1749000&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=A1991GV58300028&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-27-1" title="View reference  in text"
                           id="ref-27">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.27">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Hong</span>  <span class="cit-name-given-names">EJ</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Rotamer optimization for protein design through MAP estimation and problem-size reduction</span>. <abbr class="cit-jnl-abbrev">J. Comput. Chem.</abbr> <span class="cit-pub-date">2009</span>;<span class="cit-vol">30</span>:<span class="cit-fpage">1923</span>-<span class="cit-lpage">1945</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/jcc.21188&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=19123203&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-28-1" title="View reference  in text"
                           id="ref-28">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.28">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Hornak</span>  <span class="cit-name-given-names">V</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Comparison of multiple Amber force fields and development of improved protein backbone parameters</span>. <abbr class="cit-jnl-abbrev">Proteins</abbr> <span class="cit-pub-date">2006</span>;<span class="cit-vol">65</span>:<span class="cit-fpage">712</span>-<span class="cit-lpage">725</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/prot.21123&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=16981200&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000241247100017&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-29-1" title="View reference  in text"
                           id="ref-29">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.29">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Humphris</span>  <span class="cit-name-given-names">EL</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Kortemme</span>  <span class="cit-name-given-names">T</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Prediction of protein-protein interface sequence diversity using flexible backbone computational protein design</span>. <abbr class="cit-jnl-abbrev">Structure</abbr> <span class="cit-pub-date">2008</span>;<span class="cit-vol">16</span>:<span class="cit-fpage">1777</span>-<span class="cit-lpage">1788</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=19081054&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-30-1" title="View reference  in text"
                           id="ref-30">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.30">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Janin</span>  <span class="cit-name-given-names">J</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Conformation of amino acid sidechains in proteins</span>. <abbr class="cit-jnl-abbrev">J. Mol. Biol.</abbr> <span class="cit-pub-date">1978</span>;<span class="cit-vol">125</span>:<span class="cit-fpage">357</span>-<span class="cit-lpage">386</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1016/0022-2836(78)90408-4&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=731698&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=A1978FY32300007&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-31-1" title="View reference  in text"
                           id="ref-31">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.31">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Kingsford</span>  <span class="cit-name-given-names">CL</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Solving and analyzing side-chain positioning problems using linear and integer programming</span>. <abbr class="cit-jnl-abbrev">Bioinformatics</abbr> <span class="cit-pub-date">2005</span>;<span class="cit-vol">21</span>:<span class="cit-fpage">1028</span>-<span class="cit-lpage">1036</span>.</cite></div>
                           <div class="cit-extra"><a href="/cgi/ijlink?linkType=ABST&amp;journalCode=bioinfo&amp;resid=21/7/1028"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-ijlinks"><span class="cit-reflinks-abstract">Abstract</span><span class="cit-sep cit-reflinks-variant-name-sep">/</span><span class="cit-reflinks-full-text"><span class="free-full-text">FREE </span>Full Text</span></a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-32-1" title="View reference  in text"
                           id="ref-32">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.32">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Kirschner</span>  <span class="cit-name-given-names">KN</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">GLYCAM06: a generalizable biomolecular force field. Carbohydrates</span>. <abbr class="cit-jnl-abbrev">J. Comput. Chem.</abbr> <span class="cit-pub-date">2008</span>;<span class="cit-vol">29</span>:<span class="cit-fpage">622</span>-<span class="cit-lpage">655</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/jcc.20820&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=17849372&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000253648500012&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-33-1" title="View reference  in text"
                           id="ref-33">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.33">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Koster</span>  <span class="cit-name-given-names">AMC</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Solving frequency assignment problems via tree-decomposition</span>. <abbr class="cit-jnl-abbrev">Electron. Notes Discrete Math.</abbr> <span class="cit-pub-date">1999</span>;<span class="cit-vol">3</span>:<span class="cit-fpage">102</span>-<span class="cit-lpage">105</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1016/S1571-0653(05)80034-4&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-34-1" title="View reference  in text"
                           id="ref-34">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.34">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Kuhlman</span>  <span class="cit-name-given-names">B</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Baker</span>  <span class="cit-name-given-names">D</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Native protein sequences are close to optimal for their structures</span>. <abbr class="cit-jnl-abbrev">Proc. Natl Acad. Sci. USA</abbr> <span class="cit-pub-date">2000</span>;<span class="cit-vol">97</span>:<span class="cit-fpage">10383</span>-<span class="cit-lpage">10388</span>.</cite></div>
                           <div class="cit-extra"><a href="/cgi/ijlink?linkType=ABST&amp;journalCode=pnas&amp;resid=97/19/10383"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-ijlinks"><span class="cit-reflinks-abstract">Abstract</span><span class="cit-sep cit-reflinks-variant-name-sep">/</span><span class="cit-reflinks-full-text"><span class="free-full-text">FREE </span>Full Text</span></a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-35-1" title="View reference  in text"
                           id="ref-35">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.35">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Larrosa</span>  <span class="cit-name-given-names">J</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Schiex</span>  <span class="cit-name-given-names">T</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Solving weighted CSP by maintaining arc consistency</span>. <abbr class="cit-jnl-abbrev">Artif. Intell.</abbr> <span class="cit-pub-date">2004</span>;<span class="cit-vol">159</span>:<span class="cit-fpage">1</span>-<span class="cit-lpage">26</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1016/j.artint.2004.05.004&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=000224534000001&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-36-1" title="View reference  in text"
                           id="ref-36">↵</a><div class="cit ref-cit ref-book" id="cit-29.17.2129.36">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Larrosa</span>  <span class="cit-name-given-names">J</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-source">Proceedings of the National Conference on Artificial Intelligence</span>. <span class="cit-publ-loc">Madison, Wisconsin, USA</span>; <span class="cit-pub-date">1998</span>. <span class="cit-article-title">Reversible DAC and other improvements for solving Max-CSP</span>; p. <span class="cit-fpage">347</span>-<span class="cit-lpage">352</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Reversible%20DAC%20and%20other%20improvements%20for%20solving%20Max-CSP&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Larrosa&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-37-1" title="View reference  in text"
                           id="ref-37">↵</a><div class="cit ref-cit ref-book" id="cit-29.17.2129.37">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Larrosa</span>  <span class="cit-name-given-names">J</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-source">Proceedings of the 19th International Joint Conference on Artificial Intelligence</span>. <span class="cit-publ-loc">Edinburgh, Scotland, UK</span>: <span class="cit-publ-name">Morgan Kaufmann Publishers Inc.</span>; <span class="cit-pub-date">2005</span>. <span class="cit-article-title">Existential arc consistency: getting closer to full arc consistency in weighted CSPs</span>; p. <span class="cit-fpage">84</span>-<span class="cit-lpage">89</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Existential%20arc%20consistency%3A%20getting%20closer%20to%20full%20arc%20consistency%20in%20weighted%20CSPs&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Larrosa&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-38-1" title="View reference  in text"
                           id="ref-38">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.38">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Leach</span>  <span class="cit-name-given-names">AR</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Exploring the conformational space of protein side chains using dead-end elimination and the A* algorithm</span>. <abbr class="cit-jnl-abbrev">Proteins</abbr> <span class="cit-pub-date">1998</span>;<span class="cit-vol">33</span>:<span class="cit-fpage">227</span>-<span class="cit-lpage">239</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/(SICI)1097-0134(19981101)33:2%3C227::AID-PROT7%3E3.0.CO;2-F&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=9779790&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000076257100007&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-39-1" title="View reference  in text"
                           id="ref-39">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.39">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Leaver-Fay</span>  <span class="cit-name-given-names">A</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">An adaptive dynamic programming algorithm for the side chain placement problem</span>. <abbr class="cit-jnl-abbrev">Pac. Symp. Biocomput.</abbr> <span class="cit-pub-date">2005</span>:<span class="cit-fpage">16</span>-<span class="cit-lpage">27</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=An%20adaptive%20dynamic%20programming%20algorithm%20for%20the%20side%20chain%20placement%20problem&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Leaver-Fay&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-40-1" title="View reference  in text"
                           id="ref-40">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.40">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Lecoutre</span>  <span class="cit-name-given-names">C</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Reasoning from last conflict (s) in constraint programming</span>. <abbr class="cit-jnl-abbrev">Artif. Intell.</abbr> <span class="cit-pub-date">2009</span>;<span class="cit-vol">173</span>:<span class="cit-fpage">1592</span>-<span class="cit-lpage">1614</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1016/j.artint.2009.09.002&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=000272304200003&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-41-1" title="View reference  in text"
                           id="ref-41">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.41">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Lippow</span>  <span class="cit-name-given-names">SM</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Computational design of antibody affinity improvement beyond in vitro maturation</span>. <abbr class="cit-jnl-abbrev">Nat. Biotechnol.</abbr> <span class="cit-pub-date">2007</span>;<span class="cit-vol">25</span>:<span class="cit-fpage">1171</span>-<span class="cit-lpage">1176</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1038/nbt1336&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=17891135&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000250226600029&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-42-1" title="View reference  in text"
                           id="ref-42">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.42">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Nestl</span>  <span class="cit-name-given-names">BM</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Recent progress in industrial biocatalysis</span>. <abbr class="cit-jnl-abbrev">Curr. Opin. Chem. Biol.</abbr> <span class="cit-pub-date">2011</span>;<span class="cit-vol">15</span>:<span class="cit-fpage">187</span>-<span class="cit-lpage">193</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1016/j.cbpa.2010.11.019&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=21195018&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-43-1" title="View reference  in text"
                           id="ref-43">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.43">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Pabo</span>  <span class="cit-name-given-names">C</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Molecular technology: designing proteins and peptides</span>. <abbr class="cit-jnl-abbrev">Nature</abbr> <span class="cit-pub-date">1983</span>;<span class="cit-vol">301</span>:<span class="cit-fpage">200</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1038/301200a0&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=6823300&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-44-1" title="View reference  in text"
                           id="ref-44">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.44">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Pierce</span>  <span class="cit-name-given-names">NA</span></span>, 
                                 </li>
                                 <li><span class="cit-auth"><span class="cit-name-surname">Winfree</span>  <span class="cit-name-given-names">E</span></span></li>
                              </ol><cite>. <span class="cit-article-title">Protein design is NP-hard</span>. <abbr class="cit-jnl-abbrev">Protein Eng.</abbr> <span class="cit-pub-date">2002</span>;<span class="cit-vol">15</span>:<span class="cit-fpage">779</span>-<span class="cit-lpage">782</span>.</cite></div>
                           <div class="cit-extra"><a href="/cgi/ijlink?linkType=ABST&amp;journalCode=proeng&amp;resid=15/10/779"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-ijlinks"><span class="cit-reflinks-abstract">Abstract</span><span class="cit-sep cit-reflinks-variant-name-sep">/</span><span class="cit-reflinks-full-text"><span class="free-full-text">FREE </span>Full Text</span></a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-45-1" title="View reference  in text"
                           id="ref-45">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.45">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Raha</span>  <span class="cit-name-given-names">K</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Prediction of amino acid sequence from structure</span>. <abbr class="cit-jnl-abbrev">Protein Sci.</abbr> <span class="cit-pub-date">2000</span>;<span class="cit-vol">9</span>:<span class="cit-fpage">1106</span>-<span class="cit-lpage">1119</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10892804&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000087856300007&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-46-1" title="View reference  in text"
                           id="ref-46">↵</a><div class="cit ref-cit ref-book" id="cit-29.17.2129.46">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Schiex</span>  <span class="cit-name-given-names">T</span></span></li>
                              </ol><cite>. <span class="cit-source">Proceedings of Principles and Practice of Constraint Programming–CP 2000, Singapore</span>. <span class="cit-pub-date">2000</span>. <span class="cit-article-title">Arc consistency for soft constraints</span>; p. <span class="cit-fpage">411</span>-<span class="cit-lpage">425</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Arc%20consistency%20for%20soft%20constraints&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Schiex&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-47-1" title="View reference  in text"
                           id="ref-47">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.47">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Schiex</span>  <span class="cit-name-given-names">T</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Valued constraint satisfaction problems: hard and easy problems</span>. <abbr class="cit-jnl-abbrev">Int. Joint Conf. Artif. Intell.</abbr> <span class="cit-pub-date">1995</span>;<span class="cit-vol">14</span>:<span class="cit-fpage">631</span>-<span class="cit-lpage">639</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Valued%20constraint%20satisfaction%20problems%3A%20hard%20and%20easy%20problems&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Schiex&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-48-1" title="View reference  in text"
                           id="ref-48">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.48">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Voigt</span>  <span class="cit-name-given-names">CA</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Trading accuracy for speed: a quantitative comparison of search algorithms in protein sequence design</span>. <abbr class="cit-jnl-abbrev">J. Mol. Biol.</abbr> <span class="cit-pub-date">2000</span>;<span class="cit-vol">299</span>:<span class="cit-fpage">789</span>-<span class="cit-lpage">803</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1006/jmbi.2000.3758&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=10835284&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000087540100018&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-49-1" title="View reference  in text"
                           id="ref-49">↵</a><div class="cit ref-cit ref-book" id="cit-29.17.2129.49">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Wallace</span>  <span class="cit-name-given-names">RJ</span></span></li>
                              </ol><cite>. <span class="cit-source">Proceedings of the thirteenth national conference on Artificial Intelligence (AAAI-96)</span>. <span class="cit-publ-loc">Portland, Oregon</span>; <span class="cit-pub-date">1996</span>. <span class="cit-article-title">Enhancements of branch and bound methods for the maximal constraint satisfaction problem</span>; p. <span class="cit-fpage">188</span>-<span class="cit-lpage">195</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Enhancements%20of%20branch%20and%20bound%20methods%20for%20the%20maximal%20constraint%20satisfaction%20problem&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Wallace&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-50-1" title="View reference  in text"
                           id="ref-50">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.50">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Wang</span>  <span class="cit-name-given-names">J</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Development and testing of a general AMBER force field</span>. <abbr class="cit-jnl-abbrev">J. Comp. Chem.</abbr> <span class="cit-pub-date">2004</span>;<span class="cit-vol">25</span>:<span class="cit-fpage">1157</span>-<span class="cit-lpage">1174</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1002/jcc.20035&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-51-1" title="View reference  in text"
                           id="ref-51">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.51">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Wang</span>  <span class="cit-name-given-names">J</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Automatic atom type and bond type perception in molecular mechanical calculations</span>. <abbr class="cit-jnl-abbrev">J. Mol. Graph. Model.</abbr> <span class="cit-pub-date">2006</span>;<span class="cit-vol">25</span>:<span class="cit-fpage">247260</span>.</cite></div>
                           <div class="cit-extra"><a target="_blank"
                                 href="http://scholar.google.com.gate1.inist.fr/scholar?as_q=&amp;as_epq=Automatic%20atom%20type%20and%20bond%20type%20perception%20in%20molecular%20mechanical%20calculations&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=Wang&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;btnG=&amp;hl=en&amp;sciui=1&amp;as_sdt=0%2C5">
                                 Search Google Scholar
                                 </a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-52-1" title="View reference  in text"
                           id="ref-52">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.52">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Wernisch</span>  <span class="cit-name-given-names">L</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">Automatic protein design with all atom force fields by exact and heuristic optimization</span>. <abbr class="cit-jnl-abbrev">J. Mol. Biol.</abbr> <span class="cit-pub-date">2000</span>;<span class="cit-vol">301</span>:<span class="cit-fpage">713</span>-<span class="cit-lpage">736</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1006/jmbi.2000.3984&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=10966779&amp;link_type=MED"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-medline">Medline</a><a href="/external-ref?access_num=000088881000010&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                     <li><span class="ref-label ref-label-empty"></span><a class="rev-xref-ref" href="#xref-ref-53-1" title="View reference  in text"
                           id="ref-53">↵</a><div class="cit ref-cit ref-journal" id="cit-29.17.2129.53">
                           <div class="cit-metadata">
                              <ol class="cit-auth-list">
                                 <li><span class="cit-auth"><span class="cit-name-surname">Zytnicki</span>  <span class="cit-name-given-names">M</span></span>, 
                                 </li>
                                 <li><span class="cit-etal">et al</span></li>
                              </ol><cite>. <span class="cit-article-title">DARN! A weighted constraint solver for RNA motif localization</span>. <abbr class="cit-jnl-abbrev">Constraints</abbr> <span class="cit-pub-date">2008</span>;<span class="cit-vol">13</span>:<span class="cit-fpage">91</span>-<span class="cit-lpage">109</span>.</cite></div>
                           <div class="cit-extra"><a href="/external-ref?access_num=10.1007/s10601-007-9033-9&amp;link_type=DOI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">CrossRef</a><a href="/external-ref?access_num=000255408000005&amp;link_type=ISI"
                                 class="cit-ref-sprinkles cit-ref-sprinkles-webofscience">Web of Science</a></div>
                        </div>
                     </li>
                  </ol>
               </div><span class="highwire-journal-article-marker-end"></span></div><span id="related-urls"></span></div>
         <div id="col-2">
            
            <div class="article-nav sidebar-nav">
               <a href="/content/29/17/2122.short" title="Previous article" class="previous">« Previous</a><span class="article-nav-sep"> | </span><a href="/content/29/17/2137.short" title="Next article" class="next">Next Article »</a>
               
               <span class="toc-link">
                  				<a href="/content/29/17.toc" title="Table of Contents">Table of Contents</a>
                  			</span>
               
               
            </div>
            <div class="content-box" id="article-cb-main">
               <div class="cb-contents">
                  <h3 class="cb-contents-header"><span>This Article</span></h3>
                  <div class="cb-section cb-slug">
                     <ol>
                        <li>
                           <div id="slugline">
                              
                              <cite>
                                 <abbr title="Bioinformatics" class="slug-jnl-abbrev">
                                    <nlm:abbrev-journal-title xmlns:nlm="http://schema.highwire.org.gate1.inist.fr/NLM/Journal" abbrev-type="publisher">Bioinformatics</nlm:abbrev-journal-title></abbr><span class="slug-pub-date" itemprop="datePublished">
                                    (2013) 
                                    </span>
                                 
                                 <span class="slug-vol">
                                    29
                                    </span><span class="slug-issue">
                                    (17):
                                    </span><span class="slug-pages">
                                    2129-2136.
                                    </span>
                                 <span class="slug-doi-wrapper">
                                    
                                    doi:
                                    
                                    <span title="10.1093/bioinformatics/btt374" class="slug-doi">10.1093/bioinformatics/btt374</span>
                                    </span>
                                 </cite>
                              
                              
                              <span class="slug-metadata-note ahead-of-print">
                                 
                                 First published online:
                                 
                                 <span class="slug-ahead-of-print-date">July 10, 2013</span>
                                 </span>
                              
                              
                              
                              
                              
                              
                              
                              
                              
                           </div>
                        </li>
                     </ol>
                  </div>
                  <div class="cb-section cb-views">
                     <ol>
                        <li class="abstract-view-link primary"><a href="/content/29/17/2129.abstract" rel="view-abstract">Abstract</a><span class="viewspecificaccesscheck bioinfo;29/17/2129 abstract"></span></li>
                        <li class="notice full-text-view-link"><span class="variant-indicator">» <span>Full Text (HTML)</span></span><span class="viewspecificaccesscheck bioinfo;29/17/2129 full"></span></li>
                        <li class="notice full-text-pdf-view-link"><a href="/content/29/17/2129.full.pdf+html" rel="view-full-text.pdf">Full Text (PDF)</a><span class="viewspecificaccesscheck bioinfo;29/17/2129 reprint"></span></li>
                        <li><a href="/content/29/17/2129/suppl/DC1" rel="supplemental-data"
                              class="dslink-supplementary-data">Supplementary Data</a></li>
                        <li class="cb-versions">
                           <div><span id="pap-all-versions-label">All Versions of this Article:</span><ol class="version-list">
                                 <li><a href="/content/early/2013/07/10/bioinformatics.btt374">btt374v1</a></li>
                                 <li><a href="/content/early/2013/07/19/bioinformatics.btt374">btt374v2</a></li>
                                 <li class="current-li"><span>29/17/2129</span> <span class="current-version">most recent</span></li>
                              </ol>
                           </div>
                        </li>
                     </ol>
                  </div>
                  <div class="cb-section collapsible" id="cb-art-cat">
                     <h4 class="cb-section-header"><span>Classifications</span></h4>
                     <ol>
                        <li>
                           <ul class="subject-headings">
                              <li>Original Paper
                                 <ul class="subject-sub-headings last-child">
                                    <li><a class="tocsection-search"
                                          href="/search?tocsectionid=STRUCTURAL+BIOINFORMATICS&amp;sortspec=date&amp;submit=Submit">STRUCTURAL BIOINFORMATICS</a></li>
                                 </ul>
                              </li>
                           </ul>
                        </li>
                     </ol>
                  </div>
                  <div class="cb-section collapsible" id="cb-art-svcs">
                     <h4 class="cb-section-header"><span>Services</span></h4>
                     <ol>
                        <li class="alert-link icon-link"><a href="/cgi/alerts/ctalert?alertType=citedby&amp;addAlert=cited_by&amp;cited_by_criteria_resid=bioinfo;29/17/2129&amp;saveAlert=no&amp;return-type=article&amp;return_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129.full">Alert me when cited</a></li>
                        <li class="alert-link icon-link"><a href="/cgi/alerts/ctalert?alertType=correction&amp;addAlert=correction&amp;correction_criteria_value=29/17/2129&amp;saveAlert=no&amp;return-type=article&amp;return_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129.full">Alert me if corrected</a></li>
                        <li><a href="/cgi/alerts/ctalert?alertType=eletter&amp;addAlert=eletter&amp;eletter_criteria_value=29/17/2129&amp;saveAlert=no&amp;return-type=article&amp;return_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129.full">Alert me if commented</a></li>
                        <li class="similar-link"><a href="/search?qbe=bioinfo;btt374&amp;citation=Traor%C3%A9%20et%20al.%2029%20%2817%29:%202129&amp;submit=yes">Find similar articles</a></li>
                        <li class="similar-link"><a href="/external-ref?access_num=bioinfo%3B29%2F17%2F2129&amp;link_type=ISI_RELATEDRECORDS"
                              id="cb-isi-similar-articles"
                              class="similar-link">Similar articles in Web of Science</a></li>
                        <li class="similar-link"><a href="/external-ref?access_num=23842814&amp;link_type=MED_NBRS"
                              class="similar-link">Similar articles in PubMed</a></li>
                        <li class="folder-link icon-link"><a href="/cgi/folders?action=addtofolder&amp;wherefrom=JOURNALS&amp;wrapped_id=bioinfo;29/17/2129">Add to my archive</a></li>
                        <li class="cit-man-link icon-link cite-link"><a href="/citmgr?gca=bioinfo;29/17/2129">Download citation</a></li>
                        <li><a class="request-permissions"
                              href="https://s100.copyright.com/AppDispatchServlet?publisherName=oup&amp;publication=bioinformatics&amp;title=A%20new%20framework%20for%20computational%20protein%20design%20through%20cost%20function%20network%20optimization%3A%20&amp;publicationDate=09%2F01%2F2013&amp;author=Seydou%20Traor%C3%A9%2C%20David%20Allouche%2C%20Isabelle%20Andr%C3%A9%2C%20Simon%20de%20Givry%2C%20George%20Katsirelos%2C%20Thomas%20Schiex%2C%20Sophie%20Barbe&amp;copyright=Oxford%20University%20Press&amp;contentID=10.1093%2Fbioinformatics%2Fbtt374&amp;volumeNum=29&amp;issueNum=17&amp;startPage=2129&amp;endPage=2136&amp;orderBeanReset=true"
                              target="_blank">Request Permissions</a></li>
                     </ol>
                  </div>
                  <div class="cb-section collapsible default-closed" id="cb-letter-cit">
                     <h4 class="cb-section-header"><span>Responses</span></h4>
                     <ol>
                        <li><a href="/letters/submit/bioinfo;29/17/2129">Submit a response</a></li>
                        <li><span class="eletters-no-responses-published">No responses published</span></li>
                     </ol>
                  </div>
                  <div class="cb-section collapsible default-closed" id="cb-art-cit">
                     <h4 class="cb-section-header"><span>Citing Articles</span></h4>
                     <ol>
                        <li class="hw-citing-link"><a href="/content/29/17/2129.full?cited-by=yes&amp;legid=bioinfo;29/17/2129#cited-by"
                              id="cb-hw-citing-articles">Load citing article information</a></li>
                        <li><a href="/cgi/crossref-forward-links/29/17/2129" rel="external-nw"
                              id="cb-crossref-citing-articles">Citing articles via CrossRef</a></li>
                        <li><a href="/scopus-links/bioinfo/29/17/2129?access_num=/bioinfo/29/17/2129"
                              rel="external-nw"
                              id="cb-scopus-citing-articles">Citing articles via Scopus</a></li>
                        <li><a href="/external-ref?access_num=%2Fbioinfo%2F29%2F17%2F2129&amp;link_type=ISI_CITING&amp;accnum_type=native"
                              rel="external-nw"
                              id="cb-isi-citing-articles">Citing articles via Web of Science</a></li>
                        <li><a href="/external-ref?access_num=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129.abstract&amp;link_type=GOOGLESCHOLAR">Citing articles via Google Scholar</a></li>
                     </ol>
                  </div>
                  <div class="cb-section collapsible default-closed" id="cb-art-gs">
                     <h4 class="cb-section-header"><span>Google Scholar</span></h4>
                     <ol>
                        <li class="cb-art-gs-auth author-link"><a href="http://scholar.google.com.gate1.inist.fr/scholar?q=%22author%3ATraor%C3%A9%20author%3AS.%22"
                              class="cb-art-gs-auth author-link">Articles by  Traoré, S.</a></li>
                        <li class="cb-art-gs-auth author-link"><a href="http://scholar.google.com.gate1.inist.fr/scholar?q=%22author%3ABarbe%20author%3AS.%22"
                              class="cb-art-gs-auth author-link">Articles by  Barbe, S.</a></li>
                        <li class="cb-art-gs-rel similar-link"><a href="/external-ref?access_num=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/content/29/17/2129.abstract&amp;link_type=GOOGLESCHOLARRELATED">Search for related content</a></li>
                     </ol>
                  </div>
                  <div class="cb-section collapsible default-closed" id="cb-art-pm">
                     <h4 class="cb-section-header"><span>PubMed</span></h4>
                     <ol>
                        <li class="cb-art-pm-cite cite-link icon-link"><a href="/external-ref?access_num=23842814&amp;link_type=PUBMED"
                              class="cb-art-pm-cite cite-link icon-link">PubMed citation</a></li>
                        <li class="cb-art-pm-auth author-link"><a href="/external-ref?access_num=Traor%C3%A9%20S&amp;link_type=AUTHORSEARCH"
                              class="cb-art-pm-auth author-link">Articles by  Traoré, S.</a></li>
                        <li class="cb-art-pm-auth author-link"><a href="/external-ref?access_num=Allouche%20D&amp;link_type=AUTHORSEARCH"
                              class="cb-art-pm-auth author-link">Articles by  Allouche, D.</a></li>
                        <li class="cb-art-pm-auth author-link"><a href="/external-ref?access_num=Andr%C3%A9%20I&amp;link_type=AUTHORSEARCH"
                              class="cb-art-pm-auth author-link">Articles by  André, I.</a></li>
                        <li class="cb-art-pm-auth author-link"><a href="/external-ref?access_num=de%20Givry%20S&amp;link_type=AUTHORSEARCH"
                              class="cb-art-pm-auth author-link">Articles by  de Givry, S.</a></li>
                        <li class="cb-art-pm-auth author-link"><a href="/external-ref?access_num=Katsirelos%20G&amp;link_type=AUTHORSEARCH"
                              class="cb-art-pm-auth author-link">Articles by  Katsirelos, G.</a></li>
                        <li class="cb-art-pm-auth author-link"><a href="/external-ref?access_num=Schiex%20T&amp;link_type=AUTHORSEARCH"
                              class="cb-art-pm-auth author-link">Articles by  Schiex, T.</a></li>
                        <li class="cb-art-pm-auth author-link"><a href="/external-ref?access_num=Barbe%20S&amp;link_type=AUTHORSEARCH"
                              class="cb-art-pm-auth author-link">Articles by  Barbe, S.</a></li>
                     </ol>
                  </div>
                  <div class="cb-section collapsible default-closed" id="cb-art-rel">
                     <h4 class="cb-section-header"><span>Related Content</span></h4>
                     <ol>
                        <li><a href="/content/29/17/2129.full?related-urls=yes&amp;legid=bioinfo;29/17/2129#related-urls"
                              id="cb-related-urls">Load related web page information</a></li>
                     </ol>
                  </div>
                  <div class="cb-section collapsible" id="cb-art-soc">
                     <h4 class="cb-section-header"><span>Share</span></h4>
                     <ol>
                        <li><a href="/email?gca=bioinfo;29/17/2129&amp;current-view-path=/content/29/17/2129.full">
                              Email this article</a></li>
                        <li>
                           <div class="social-bookmarking">
                              
                              <ul class="social-bookmark-links">
                                 <li class="social-bookmarking-item social-bookmarking-item-citeulike"><a href="/external-ref?tag_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/long/29/17/2129&amp;title=A%20new%20framework%20for%20computational%20protein%20design%20through%20cost%20function%20network%20optimization+--+Traor%C3%A9%20et%20al.%2029%20%2817%29%3A%202129+--+bioinformatics&amp;doi=10.1093/bioinformatics/btt374&amp;link_type=CITEULIKE"><img src="/shared/img/common/social-bookmarking/citeulike.gif"
                                            alt="Add to CiteULike"
                                            title="CiteULike" /></a><span class="soc-bm-link-text">CiteULike</span></li>
                                 <li class="social-bookmarking-item social-bookmarking-item-delicious"><a href="/external-ref?tag_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/long/29/17/2129&amp;title=A%20new%20framework%20for%20computational%20protein%20design%20through%20cost%20function%20network%20optimization+--+Traor%C3%A9%20et%20al.%2029%20%2817%29%3A%202129+--+bioinformatics&amp;doi=10.1093/bioinformatics/btt374&amp;link_type=DEL_ICIO_US"><img src="/shared/img/common/social-bookmarking/delicious.gif"
                                            alt="Add to Delicious"
                                            title="Delicious" /></a><span class="soc-bm-link-text">Delicious</span></li>
                                 <li class="social-bookmarking-item social-bookmarking-item-facebook"><a href="/external-ref?tag_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/29/17/2129&amp;title=A%20new%20framework%20for%20computational%20protein%20design%20through%20cost%20function%20network%20optimization+--+Traor%C3%A9%20et%20al.%2029%20%2817%29%3A%202129+--+bioinformatics&amp;doi=10.1093/bioinformatics/btt374&amp;link_type=FACEBOOK"
                                       class="sb-facebook"
                                       rel="external-nw"><img src="/shared/img/common/social-bookmarking/facebook.gif" alt="Add to Facebook"
                                            title="Facebook" /></a><span class="soc-bm-link-text">Facebook</span></li>
                                 <li class="social-bookmarking-item social-bookmarking-item-googleplus"><a href="/external-ref?tag_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/long/29/17/2129&amp;title=A%20new%20framework%20for%20computational%20protein%20design%20through%20cost%20function%20network%20optimization+--+Traor%C3%A9%20et%20al.%2029%20%2817%29%3A%202129+--+bioinformatics&amp;doi=10.1093/bioinformatics/btt374&amp;link_type=GOOGLEPLUS&amp;log_only=yes"><img src="/shared/img/common/social-bookmarking/googleplus.jpg" alt="Add to Google+"
                                            title="Google+" /></a><span class="soc-bm-link-text">Google+</span></li>
                                 <li class="social-bookmarking-item social-bookmarking-item-mendeley"><a href="/external-ref?tag_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/long/29/17/2129&amp;title=A%20new%20framework%20for%20computational%20protein%20design%20through%20cost%20function%20network%20optimization+--+Traor%C3%A9%20et%20al.%2029%20%2817%29%3A%202129+--+bioinformatics&amp;doi=10.1093/bioinformatics/btt374&amp;link_type=MENDELEY"><img src="/shared/img/common/social-bookmarking/mendeley.gif" alt="Add to Mendeley"
                                            title="Mendeley" /></a><span class="soc-bm-link-text">Mendeley</span></li>
                                 <li class="social-bookmarking-item social-bookmarking-item-twitter"><a href="/external-ref?tag_url=http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/long/29/17/2129&amp;title=A%20new%20framework%20for%20computational%20protein%20design%20through%20cost%20function%20network%20optimization+--+Traor%C3%A9%20et%20al.%2029%20%2817%29%3A%202129+--+bioinformatics&amp;doi=10.1093/bioinformatics/btt374&amp;link_type=TWITTER"><img src="/shared/img/common/social-bookmarking/twitter.gif" alt="Add to Twitter"
                                            title="Twitter" /></a><span class="soc-bm-link-text">Twitter</span></li>
                              </ul>
                              
                              <p class="social-bookmarking-help"><a href="/help/social_bookmarks.dtl">What's this?</a></p>
                              
                           </div>
                        </li>
                     </ol>
                  </div>
               </div>
            </div>
            
            
            <div class="content-box" id="article-dyn-nav">
               <div class="cb-contents">
                  <h3 class="cb-contents-header"><span>Navigate This Article</span></h3>
                  <div class="cb-section" id="cb-art-nav">
                     <ol>
                        <li><a href="#content-block">Top</a></li>
                        <li><a href="#abstract-1">Abstract</a></li>
                        <li><a href="#sec-1">1 INTRODUCTION</a></li>
                        <li><a href="#sec-2">2 METHODS</a></li>
                        <li><a href="#sec-11">3 RESULTS AND DISCUSSION</a></li>
                        <li><a href="#sec-16">4 CONCLUSION</a></li>
                        <li><a href="#ack-1">ACKNOWLEDGEMENTS</a></li>
                        <li><a href="#fn-group-1">Footnotes</a></li>
                        <li><a href="#ref-list-1">REFERENCES</a></li>
                     </ol>
                  </div>
               </div>
            </div>
            
         </div>
         <div id="col-3">
            
            <div id="col-3-qs" class="sidebar-qs">
               
               
               <form class="searchbox" action="/search" method="get">
                  <p>Search this journal:</p>
                  <div>
                     	
                     	
                     	<input value="" type="text" name="fulltext" id="header-qs-input" />
                     <input type="hidden" name="submit" value="yes" />
                     <input class="inline_button" type="image" alt="Go" src="/publisher/img/buttons/go.gif" />
                     
                     <div class="adv-search-link"><a href="/search">Advanced »</a></div>
                     
                     
                  </div>
               </form>
               
            </div>
            <div class="cb-contents">
               
               <div id="sidebar-current-issue" class="content-box">
                  	
                  <h3 class="cb-contents-header">Current Issue</h3>
                  	
                  <div class="cb-section">
                     	  
                     <ol>
                        <li>
                           	      <a href="/content/current"><span>
                                 		  <span class="current-issue-date">
                                    March 1, 2014
                                    </span><span class="toc-citation-volume">
                                    
                                    30
                                    </span><span class="toc-citation-issue">(5)</span>
                                 		  
                                 		</span></a>
                           	    
                        </li>
                     </ol>
                     	
                  </div>
                  	
                  <div class="cb-section">
                     	  
                     <ol>
                        <li>
                           <a href="/content/current"><img src="/content/30/5.cover.gif" alt="Bioinformatics" id="cover" /></a>
                           
                        </li>
                     </ol>
                     	
                  </div>
                  	
                  <div class="cb-section sidebar-etoc-link">
                     	  
                     <ol>
                        <li><a href="/cgi/alerts/etoc">Alert me to new issues</a></li>
                     </ol>
                     	
                  </div>
                  
               </div>
               
            </div>
            
            <div id="second">
               
               <div class="features">
                  
                  <div class="feature separator_after first">
                     
                     <h2 id="the_journal" title="The Journal"><span>The Journal</span></h2>
                     
                     <ul class="emphasised">
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/bioinformatics/about.html" rel=""
                              title="About this journal">About this journal</a>
                           
                        </li>
                        <li>
                           <a href="/cgi/calendarcontent/index.html?target=hw.dtl" rel=""
                              title="Calendar of events">Calendar of events</a>
                           
                        </li>
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/access_purchase/rights_permissions.html"
                              rel=""
                              title="Rights &amp; Permissions">Rights &amp; Permissions</a>
                           
                        </li>
                        <li>
                           <a href="HTTP://www.oxfordjournals.org.gate1.inist.fr/access_purchase/dispatch_dates.html#CABIOSP"
                              rel=""
                              title="Dispatch date of the next issue">Dispatch date of the next issue</a>
                           
                        </li>
                        <li>
                           <a href="http://publicationethics.org/about" rel=""
                              title="This journal is a member of the Committee on Publication Ethics (COPE)">This journal is a member of the Committee on Publication Ethics (COPE)</a>
                           
                        </li>
                        <li>
                           <a href="/cgi/eletters?lookup=by_date&amp;days=84&amp;target=hw.dtl" rel=""
                              title="Recent Comments">Recent Comments</a>
                           
                        </li>
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/mobile_faqs.html" rel=""
                              title="We are mobile – find out more">We are mobile – find out more</a>
                           
                        </li>
                     </ul>
                     <a href="http://www.oxfordjournals.org.gate1.inist.fr/page/4638/1">
                        <img alt="Next Generation Sequencing "
                             src="http://www.oxfordjournals.org.gate1.inist.fr/our_journals/bioinformatics/nextgensuq_sidebar_2012.jpg" />
                        </a>
                     
                  </div>
                  
                  <div class="feature separator_after">
                     
                     <h3>An official journal of</h3>
                     
                     <ul class="emphasised">
                        <li>
                           <a href="http://www.iscb.org/" rel=""
                              title="The International Society for Computational Biology">The International Society for Computational Biology</a>
                           
                        </li>
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/our_journals/bioinformatics/iscb_articles.html"
                              rel=""
                              title="Click here to read ISCB articles">Click here to read ISCB articles</a>
                           
                        </li>
                     </ul>
                     
                  </div>
                  
                  <div class="feature separator_after">
                     
                     <h3>Impact factor:  5.323</h3>
                     
                     <h3 class="five_year">5-Yr impact factor: 6.911</h3>
                     
                  </div>
                  
                  <div class="feature">
                     
                     <h3>Editors-in-Chief</h3>
                     
                     <div class="contact">
                        
                        <div class="name">J Kelso</div>
                        
                        <div class="clear"></div>
                        
                     </div>
                     
                     <div class="contact">
                        
                        <div class="name">A Valencia</div>
                        
                        <div class="clear"></div>
                        
                     </div>
                     
                     <ul class="emphasised">
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/bioinformatics/editorial_board.html" rel=""
                              title="View full editorial board">View full editorial board</a>
                           
                        </li>
                     </ul>
                     
                  </div>
                  
                  <div class="feature">
                     
                     <h2 id="for_authors" title="For Authors"><span>For Authors</span></h2>
                     
                     <ul class="emphasised">
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/bioinformatics/for_authors/index.html"
                              rel=""
                              title="Instructions to authors">Instructions to authors</a>
                           
                        </li>
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/our_journals/bioinformatics/for_authors/submission_online.html"
                              rel=""
                              title="Online submission">Online submission</a>
                           
                        </li>
                        <li>
                           <a href="http://mc.manuscriptcentral.com/bioinformatics" rel=""
                              title="Submit Now!">Submit Now!</a>
                           
                        </li>
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/access_purchase/self-archiving_policyb.html"
                              rel=""
                              title="Self-archiving policy">Self-archiving policy</a>
                           
                        </li>
                     </ul>
                     <a href="http://www.oxfordjournals.org.gate1.inist.fr/oxfordopen/ ">
                        <img alt="Open access options for authors - visit Oxford Open"
                             src="http://www.oxfordjournals.org.gate1.inist.fr/resource/image/promos/oxford_open_small.gif" />
                        </a>
                     <a href="http://www.oxfordjournals.org.gate1.inist.fr/oxfordopen/funder_policies/">
                        <img alt="RCUK Wellcome Trust Open Access"
                             src="http://www.oxfordjournals.org.gate1.inist.fr/resource/image/wellcome-trust-180x100.jpg" />
                        </a>
                     
                     <p>Open access options for authors - visit <a href="http://www.oxfordjournals.org.gate1.inist.fr/oxfordopen/">Oxford Open</a>
                        
                     </p>
                     <a href="http://publicaccess.nih.gov.gate1.inist.fr/submit_process_journals.htm">
                        <img alt="PMC Logo" src="http://www.oxfordjournals.org.gate1.inist.fr/resource/image/pmc-logo.gif" />
                        </a>
                     
                     <ul class="emphasised">
                        <li>
                           <a href="http://publicaccess.nih.gov.gate1.inist.fr/submit_process_journals.htm" rel=""
                              title="This journal enables compliance with the NIH Public Access Policy">This journal enables compliance with the NIH Public Access Policy</a>
                           
                        </li>
                     </ul>
                     
                  </div>
                  
                  <div class="feature">
                     
                     <h2 id="alerting_services" title="Alerting Services"><span>Alerting Services</span></h2>
                     
                     <ul class="emphasised">
                        <li>
                           <a href="/cgi/alerts/etoc" rel="" title="Email table of contents">Email table of contents</a>
                           
                        </li>
                        <li>
                           <a href="/cgi/alerts/etoc" rel="" title="Email Advance Access">Email Advance Access</a>
                           
                        </li>
                        <li>
                           <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/help/citetrack/index.dtl" rel=""
                              title="CiteTrack">CiteTrack</a>
                           
                        </li>
                        <li>
                           <a href="/rss" rel="" title="XML RSS feed">XML RSS feed</a>
                           
                        </li>
                     </ul>
                     
                  </div>
                  
                  <div class="feature">
                     
                     <h2 id="corporate_services" title="Corporate Services"><span>Corporate Services</span></h2>
                     
                     <ul class="emphasised">
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/corporate_services/advertising.html" rel=""
                              title="Advertising sales">Advertising sales</a>
                           
                        </li>
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/corporate_services/reprints.html" rel=""
                              title="Reprints">Reprints</a>
                           
                        </li>
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/corporate_services/supplements.html" rel=""
                              title="Supplements">Supplements</a>
                           
                        </li>
                     </ul>
                     
                  </div>
                  
                  <div class="feature">
                     
                     <h2>Widget</h2>
                     
                     <ul class="emphasised">
                        <li>
                           <a href="http://www.oxfordjournals.org.gate1.inist.fr/our_journals/bioinformatics/widget.html"
                              rel=""
                              title="Get a widget">Get a widget</a>
                           
                        </li>
                     </ul>
                     
                  </div>
                  
               </div>
               
               <div class="ad_hidden" id="oas_right1">
                  <iframe id="id_advertframe_right1"
                          src="/resource/htmlfiles/advert.html?p=Right1&amp;u=bioinformatics.oxfordjournals.org/content/29/17/2129.full"
                          width="0"
                          height="0"
                          frameborder="0"
                          scrolling="no"
                          class="resize_ad"></iframe>
                  
               </div>
               <script type="text/javascript">var taxonomies = ["SCI01060"];</script>
               </div>
            
            <div class="most-links-box js-marker">
               
               <div class="most-header">
                  
                  <h3>Most</h3>
                  
               </div>
               
               <ul>
                  <li class="most-cur-sel">
                     <h4>Most Read</h4>
                     	  
                     <div class="most-list">
                        	    
                        <ol>
                           <li class="first-item"><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/30/3/301?rss=1&amp;amp;ssource=mfr">RNA-seq differential expression studies: more sequence or more replication?</a></li>
                           <li><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/30/1/38?rss=1&amp;amp;ssource=mfr">A poor man's BLASTX--high-throughput metagenomic protein database search using PAUDA</a></li>
                           <li><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/btt764v2?rss=1&amp;amp;ssource=mfr">CasOT: a genome-wide Cas9/gRNA off-target searching tool</a></li>
                           <li><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/btt756v2?rss=1&amp;amp;ssource=mfr">CGAT: computational genomics analysis toolkit</a></li>
                           <li class="last-item"><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/btt737v2?rss=1&amp;amp;ssource=mfr">WiggleTools: parallel processing of large collections of genome-wide datasets for visualization and statistical analysis</a></li>
                        </ol>
                        	    <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/reports/mfr1.dtl"
                           class="view-all">» View all Most Read articles</a>
                        	  
                     </div>
                     	
                  </li>
                  <li>
                     <h4>Most Cited</h4>
                     	
                     <div class="most-list">
                        	  
                        <ol>
                           <li class="first-item"><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/28/8/1086?rss=1&amp;amp;ssource=mfc">Oases: robust de novo RNA-seq assembly across the dynamic range of expression levels</a></li>
                           <li><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/28/11/1420?rss=1&amp;amp;ssource=mfc">IDBA-UD: a de novo assembler for single-cell and metagenomic sequencing data with highly uneven depth</a></li>
                           <li><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/28/19/2537?rss=1&amp;amp;ssource=mfc">GenAlEx 6.5: genetic analysis in Excel. Population genetic software for teaching and research--an update</a></li>
                           <li><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/28/12/1647?rss=1&amp;amp;ssource=mfc">Geneious Basic: An integrated and extendable desktop software platform for the organization and analysis of sequence data</a></li>
                           <li class="last-item"><a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/cgi/content/short/28/4/593?rss=1&amp;amp;ssource=mfc">ART: a next-generation sequencing read simulator</a></li>
                        </ol>
                        	  <a href="http://bioinformatics.oxfordjournals.org.gate1.inist.fr/reports/mfc1.dtl"
                           class="view-all">» View all Most Cited articles</a>
                        	
                     </div>
                     
                  </li>
               </ul>
               
            </div>
            
            
            
         </div>
         
         
         
         <div id="oas_bottom" class="ad_hidden">
            <iframe id="id_advertframe_bottom"
                    src="/resource/htmlfiles/advert.html?p=Bottom&amp;u=bioinformatics.oxfordjournals.org/content/29/17/2129.full"
                    width="0"
                    height="0"
                    frameborder="0"
                    scrolling="no"
                    class="resize_ad"></iframe>
            
         </div>
         
         
         
         
         
         
         <div id="secondary_footer">
            
            <div id="issn">Online ISSN 1460-2059 - Print ISSN 1367-4803</div>
            
            
            <div id="copyright"><a href="http://www.oxfordjournals.org.gate1.inist.fr/our_journals/terms.html">Copyright ©</a> 
               <span>2014</span> Oxford University Press
               
            </div>
            
         </div>
         
         <div id="primary_footer">
            
            <div id="site_logo">
               <strong><span>Oxford Journals</span></strong>
               <em><span>Oxford University Press</span></em>
               
            </div>
            
            
            <div id="third_nav">
               
               <ul>
                  <li><a href="http://www.oxfordjournals.org.gate1.inist.fr/site_map.html">Site Map</a></li>
                  <li><a href="http://www.oxfordjournals.org.gate1.inist.fr/legal/privacy_policy.html">Privacy Policy</a></li>
                  <li><a href="http://global.oup.com.gate1.inist.fr/cookiepolicy/?siteid=journals">Cookie Policy</a></li>
                  <li><a href="http://www.oxfordjournals.org.gate1.inist.fr/legal/legal_notices.html">Legal Notices</a></li>
                  <li><a href="http://www.oxfordjournals.org.gate1.inist.fr/faq/">Frequently Asked Questions</a></li>
               </ul>
               
            </div>
            
            
            <div id="other_oup_sites">
               
               <form action="http://www.oxfordjournals.org.gate1.inist.fr/service/redirect">
                  <div>
                     <label for="site_list">Other Oxford University Press sites:</label>
                     <select name="url" id="site_list">
                        <option value="http://www.oup.com.gate1.inist.fr">Oxford University Press</option>
                        <option value="http://www.oxfordjournals.org.gate1.inist.fr/china/">Oxford Journals China</option>
                        <option value="http://www.oxfordjournals.org.gate1.inist.fr/japan/">Oxford Journals Japan</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/academic">Academic &amp; Professional books</option>
                        <option value="http://www.oup.com.gate1.inist.fr/oxed">Children's &amp; Schools Books</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/dictionaries">Dictionaries &amp; Reference</option>
                        <option value="http://www.oup.co.uk.gate1.inist.fr/newdnb/">Dictionary of National Biography</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/digital_reference">Digital Reference</option>
                        <option value="http://www.oup.com.gate1.inist.fr/elt/">English Language Teaching</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/textbooks">Higher Education Textbooks</option>
                        <option value="http://www.oup.co.uk.gate1.inist.fr/ieu/">International Education Unit</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/law">Law</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/medicine">Medicine</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/music">Music</option>
                        <option value="http://www.oup.com.gate1.inist.fr/online">Online Products &amp; Publishing</option>
                        <option value="http://oxfordbibliographiesonline.com">Oxford Bibliographies Online</option>
                        <option value="http://oxforddictionaries.com">Oxford Dictionaries Online</option>
                        <option value="http://www.oed.com">Oxford English Dictionary</option>
                        <option value="http://www.oxfordlanguagedictionaries.com">Oxford Language Dictionaries Online</option>
                        <option value="http://www.oxfordscholarship.com">Oxford Scholarship Online</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/reference">Reference</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/rights">Rights and Permissions</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/booksellers">Resources for Retailers &amp; Wholesalers</option>
                        <option value="http://www.oup.com.gate1.inist.fr/uk/corporate">Resources for the Healthcare Industry</option>
                        <option value="http://www.oup.com.gate1.inist.fr/vsi">Very Short Introductions</option>
                        <option value="http://www.oup.com.gate1.inist.fr/worldsclassics">World's Classics</option></select>
                     
                     <input type="image" src="/resource/image/buttons/go.gif" alt="Go"
                            class="inline_button" />
                     
                  </div>
               </form>
               
            </div>
            
         </div>
         <script type="text/javascript"
                 src="http://gab.cookie.oup.com.gate1.inist.fr/aws-cookie/oup.cookiepolicy.pack.js"
                 defer="defer"></script>
         <script type="text/javascript" src="/publisher/js/hw-publisher-baynote.js"
                 defer="defer"></script>
         <script defer="defer"
                 src="http://oi-underbar.ifactory.com/underbar/js/jquery-ui.min.js"
                 type="text/javascript"></script>
         <script defer="defer"
                 src="http://oi-underbar.ifactory.com/underbar/js/jcarousellite.js"
                 type="text/javascript"></script>
         <script defer="defer"
                 src="http://oi-underbar.ifactory.com/underbar/js/pf_oiunderbarinit.js"
                 type="text/javascript"></script>
         <script defer="defer" src="/resource/js/underbar_local.js" type="text/javascript"></script>
         <script defer="defer" type="text/javascript">
    function fnc_onDomLoaded() {
        var query_context = getQueryContext();
        PF_initOIUnderbar(query_context,":QS:default","","JRN"); 
        PF_insertOIUnderbar(0);
    };
    if (window.addEventListener) { 
        window.addEventListener('load', fnc_onDomLoaded, false); 
    } else if (window.attachEvent) { 
        window.attachEvent('onload', fnc_onDomLoaded); 
    } 
</script>
         <noscript><img src="//ouptag.scholarlyiq.com/ntpagetag.gif?js=0" height="1" width="1"
                 border="0"
                 hspace="0"
                 vspace="0"
                 alt="" /></noscript><script type="text/javascript" src="//ouptag.scholarlyiq.com/ntpagetag.js"></script><script type="text/javascript">
        var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
        document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
        </script><script type="text/javascript">
        try {
        var pageTracker = _gat._getTracker("UA-189672-16");
        pageTracker._setDomainName(".oxfordjournals.org");
        pageTracker._trackPageview();
        } catch(err) {}
        </script></div>
   </body>
</html>